

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>2. Introduction to Sequedex &mdash; Sequedex 0.9.2 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="Sequedex 0.9.2 documentation" href="index.html" />
    <link rel="next" title="3. Installation instructions" href="installation.html" />
    <link rel="prev" title="1. Copyright notice" href="copyright.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="installation.html" title="3. Installation instructions"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="copyright.html" title="1. Copyright notice"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="introduction-to-sequedex">
<h1>2. Introduction to Sequedex<a class="headerlink" href="#introduction-to-sequedex" title="Permalink to this headline">¶</a></h1>
<p>Sequedex has a public repository, wiki, and mailing list on GitHub at <a class="reference external" href="http://github.com/joelb123/Sequedex">http://github.com/joelb123/Sequedex</a>.  In the repository will be placed analysis scripts, documentation, and results of Sequedex runs on other data sets that you may compare your data to.  Sequedex users should check the site from time to time.  The html version of the documentations contains numerous web links and downloadable files.</p>
<div class="section" id="what-is-sequedex">
<h2>2.1. What is Sequedex?<a class="headerlink" href="#what-is-sequedex" title="Permalink to this headline">¶</a></h2>
<p>Sequedex is a software package for genomic analysis that was developed by an interdisciplinary team of scientists at Los Alamos National Laboratory with funding from the Laboratory-Directed Research and Development (LDRD) program.  It is currently available for use with a free two month demo license from the Technology Transfer Division of Los Alamos National Laboratory at <a class="reference external" href="http://sequedex.lanl.gov">http://sequedex.lanl.gov</a>.  The software is under active development, and information about future releases will be made available at this website, and by an email distribution to the list of users downloading the software.  A publication describing the method behind Sequedex, its underlying scientific basis in evolutionary theory and protein structure, its validation on synthetic metagenomic data, and its application to ten soil metagenomics datasets has been published. <a class="footnote-reference" href="#i1" id="id1">[1]</a>  It was a recepient of a 2012 R&amp;D 100 award. <a class="footnote-reference" href="#i2" id="id2">[2]</a></p>
<p>Sequedex currently produces phylogenetic and functional profiles of genomic and metagenomic data.  In the near future, additional functionality will be offered, including an SQLite database of classified reads that can later be filtered and extracted to a FASTA file and a more complete set of analysis scripts.</p>
<p>Sequedex classifies short genomic fragments according to phylogeny and function.  It reads nucleotide sequence data from fasta files and outputs phylogenetic profiles defined by a reference tree, functional profiles defined by example sequences in a reference classification such as the SEED project <a class="footnote-reference" href="#i6" id="id3">[3]</a> , and a profile by phylogeny and function together.</p>
</div>
<div class="section" id="how-is-sequedex-different-from-other-sequence-analysis-packages">
<h2>2.2. How is Sequedex different from other sequence analysis packages?<a class="headerlink" href="#how-is-sequedex-different-from-other-sequence-analysis-packages" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Sequedex combines a firm grounding in evolutionary theory and phylogenetic analysis with search-engine technologies.</li>
<li>Sequedex replaces the question, &#8216;How similar is this nucleotide sequence to those in a reference database?&#8217; with the question &#8216;Where has this peptide signature been observed before?&#8217;.</li>
<li>Sequedex tests phylogenetic specificity on a signature-by-signature basis, and is thus less sensitive to the confounding effects of domain swapping, paralogs, and horizontal gene transfer.</li>
<li>Sequedex relys on an explicit RNA-polymerase based phylogeny to separate the effects of inheritence from functional pressure when interpreting genomic data.</li>
<li>Sequedex can subsume functional classification schemes merely by searching example gene sets comprising a functional classification with phylogenetically-derived signature lists.</li>
<li>Sequedex classifies gene fragments at a rate of 6.6 Gbp/hour, 250,000 times faster than BLASTX against NCBI&#8217;s NR database on soil metagenomics data and runs on a laptop.</li>
</ul>
</div>
<div class="section" id="who-uses-sequedex">
<h2>2.3. Who uses Sequedex?<a class="headerlink" href="#who-uses-sequedex" title="Permalink to this headline">¶</a></h2>
<p>Although the Sequedex software package was developed specifically to analyze large shotgun metagenomics data sets characterizing bacterial communities living in soils, just about anyone who generates or analyzes genomic data, or studies systems where microbial communities are important is a potential user of Sequedex.  Specifically:</p>
<p>Sequedex is written in Java and designed to run on modern laptop or desktop computers running Linux, Windows, or Macintosh operating systems</p>
<p>Sequedex is designed to run on computers with a moderate amount of memory (6 GB of RAM for the Bact403 data module) that are readily available to data generators, who best understand the metadata, sample preparation protocols, and expected relationship of the data to previously published studies.</p>
<p>In the process of developing Sequedex, the authors have collaborated with microbial ecologists, microbiologists, evolutionary biologists, medical doctors, epidemiologists, biochemists, and cellular biologists, exploring applications such as profiling microbial communities and microbiomes, identifying attributable changes in microbial communities and microbiomes with condition, enzyme mining, pathogen diagnosis, transcriptomic and proteomic analysis of different strains of algae in differential conditions, and forensic science.  In each case, we have found Sequedex to compare favorable to existing methods in the hands of research scientists at the forefront of their field.</p>
<p>The developers have used Sequedex on all 48 cores of a modern server, to effectively run large datasets such as those derived from the human microbiome project or the diversity of environmental microbiomes deposited in NCBI&#8217;s short read archive.  Because of its speed and robust read-by-read classification on a phylogenetic tree, the ability of users to compare their samples to archival data is limited only by their willingness to download, curate, and understand reference datasets.  Even with the simple distance metrics described below show that meaningful comparisons across research groups, sequencing methodologies, and ecosystems can be made, in order to understand the phylogeny and function of metagenomics samples in comparison to existing datasets.  We thus also expect Sequedex to be useful to genomic data centers as they struggle to organize the flood of genomic data into useful ecological, medical, and microbiological conclusions.</p>
</div>
<div class="section" id="how-does-sequedex-work">
<h2>2.4. How does Sequedex work?<a class="headerlink" href="#how-does-sequedex-work" title="Permalink to this headline">¶</a></h2>
<p>The Sequedex software distribution contains a Java executable that rapidly searches genomic fragments against a classified signature list, and a data module consisting of a signature list, phylogenetic tree, and a functional classification.  Sequedex first reads the signature list into a hash map, then sequentially matches each read of an input fasta file against this list, generating an output consists of three output files: a phylogenetic profile, a functional profile, and a matrix consisting of the number of reads tallied by both function and phylogeny.</p>
<p>The specific algorithm by which reads are assigned phylogeny and function is diagramed below.  Each signature is placed on the tree at the most specific node covering all leaves at which the signature was observed.  For example, if a particular 10-mer is observed in the genomes of the seven taxa indicated by blue dots in the figure below, that signature is assigned to the node indicated by the arrow.  A similar algorithm, the least common ancestor, is employed for the frequent case of multiple signature peptides occurring in a single metagenomic read.  Each read is placed on the tree at the most specific node consistent with the signatures contained in the read.  For example, if a read contains signatures assigned to the seven nodes indicated in red on the figure below, the read is placed at the node indicated by the arrow.  If conflicting specific signatures are contained in a read, it is assigned at the most specific node covering both conflicting specific assignments.</p>
<div class="figure">
<img alt="_images/sig_assign.png" src="_images/sig_assign.png" style="width: 60%;" />
<p class="caption">Illustration of the 402-node bacterial reference tree derived from an alignment of the RNA polymerase beta and beta prime subunits, with the signature-assignment algorithm illustrated on the left in blue, and read-placement algorithm illustrated on the right in red.</p>
</div>
<p>Phylogenetic placement can be confounded by the ubiquitous processes of gene duplication, domain swapping, and horizontal gene transfer, as well as the differing gene inventories of even closely related bacteria.  By choosing a signature-by-signature placement on the phylogenetic tree, we are eliminating the ortholog-identification step from the phylogenetic profiling process.  In essence, our approach replaces the question of gene sequence similarity with the question, &#8220;Where have these signature peptides been seen before?&#8221;.  Since some signatures appear in dozens of organism, our decision to place the signature peptide close enough to the root of the tree to cover <em>every</em> observed instance of the signature can ve viewed as quite conservative.  A specific phylogenetic assignment will only be made if <strong>no</strong> conflicting evidence is available, so observed phylogenetic signals represent self-consistency in our assumptions.</p>
<p>By selecting the 5% of 10-mer peptides in our 403 reference genomes, we have identified the 20 million peptides most-likely to appear in bacteria not in our reference set.  In order to assign function to each signature peptide, we simply search example protein sequences for each of the signature peptides.  The SEED classification <a class="footnote-reference" href="#i6" id="id4">[3]</a> of protein function provides such lists for 1000 distinct classifications, as well as a hierarchical rollup to 28 different types of families.  We were able to make functional assignments for 70% of our phylogenetically-derived signature peptides in this manner.</p>
<p>Because each signature possesses both a phylogenetic and a functional assignment, it is possible to profile microbial communities on the basis of both phylogenetic and functional matches, where each read gets one count in the profile.  By design, these profiles are not biased by the depth of sequencing, assembly, or the quality of annotation of reference genomes.</p>
<div class="figure">
<img alt="_images/face_pf.png" src="_images/face_pf.png" style="width: 100%;" />
<p class="caption">Sample rollups of <strong>(left)</strong> phylogenetic and <strong>(right)</strong> functional profiles of the ten metagenomic communities.  The resolution and structure of the phylogenetic and functional classifications depend on the classification of the signature peptides. This calculation used the 402-node bacterial tree and 962-category SEED functional classification.  This data set is the one discussed at length in Reference 2.1.</p>
</div>
<p>The reference tree used for the Bact403 data module (with node numbers defined) can be downloaded in <a class="reference download internal" href="_downloads/Bacteria403.phyloxml"><tt class="xref download docutils literal"><span class="pre">phyloxml</span></tt></a>  or <a class="reference download internal" href="_downloads/Bacteria403.pdf"><tt class="xref download docutils literal"><span class="pre">pdf</span></tt></a> format.  The functional categories are described in <a class="reference download internal" href="_downloads/fn.csv"><tt class="xref download docutils literal"><span class="pre">this</span> <span class="pre">csv</span> <span class="pre">file.</span></tt></a></p>
<p>Because possibility-space for 10-mer matches is so much larger than the number of signatures, the effect of sequencing errors will typically be to cause one or more signatures in a read to not be detected.  Since this is a fairly benign mode of failure,  quality scores not used by Sequedex at present.</p>
</div>
<div class="section" id="what-is-the-specificity-and-sensitivity-of-sequedex-when-matching-bacterial-genes-and-gene-fragments-from-novel-bacteria">
<h2>2.5. What is the specificity and sensitivity of Sequedex when matching bacterial genes and gene fragments from novel bacteria?<a class="headerlink" href="#what-is-the-specificity-and-sensitivity-of-sequedex-when-matching-bacterial-genes-and-gene-fragments-from-novel-bacteria" title="Permalink to this headline">¶</a></h2>
<p>The sensitivity and specificity with which Sequedex classifies short genomic fragments is determined by the signature list, phylogenetic reference tree, and functional classification contained in the data module.  The Bact403 data module distributed with Sequedex was based on 403 reference genomes and their inferred evolutionary history, shown below and available here as a <a class="reference download internal" href="_downloads/Bacteria403.pdf"><tt class="xref download docutils literal"><span class="pre">pdf</span></tt></a>.</p>
<p>Sequedex is comparable to BLASTX in sensitivity and specificity for matching bacterial genes from phylogenetically distinct organisms, and, and is intrinsically designed to provide conservative specificity when read-assignments are made to a phylogenetic tree.  Because the signatures used for assignment by Sequedex are 30 base pairs in length, the method works nearly as well with short reads as long reads.  We employ two validation strategies here: examination of matches to well-annotated but distantly-related organisms and analysis of draft genomes of novel soil bacteria.</p>
<div class="figure">
<img alt="_images/hist_match.png" src="_images/hist_match.png" style="width: 100%;" />
<p class="caption"><strong>(left)</strong> Distribution of Protein BLAST scores (-log(E-value)) for various sets of <em>E. coli</em> genes scored against genes in the <em>B. subtilis</em> genome.  At the top, in cyan, is the distribution of the best-match BLAST scores for each of the 4145 genes in the <em>E. coli</em> genome. 1461 of these are also reciprocal best hits of the <em>B. subtilis</em> genome against <em>E. coli</em>; the distribution of these scores is shown in dark blue. 746 distinct pairs of <em>E. coli – B. subtilis</em> genes are connected by one or more <em>10-mer</em> matches; the distribution of BLAST scores for these matches is shown in red. In magenta is shown the distribution of BLAST scores for the 388 genes that are both reciprocal BLAST best hits and connected by one or more <em>10-mers</em>. At the bottom of the plot, in green, is the distribution for genes with matching <em>10-mers</em> and the word ‘transporter’ in either gene’s annotation. The peak at the right of the plot indicates the 37 pairs of genes given an E-value of ‘0.0’ by BLAST.  <strong>(right)</strong> Fraction of genes containing at least one signature peptide in genomes across the 403 bacterial reference genomes. As described in the text, signature peptides are exact matches between genomes of different genera of length 10. The genomes are ordered along the x-axis according to their position in our bacterial phylogeny.</p>
</div>
<p>The use of solid patterns of amino acids to annotate the function of proteins is startling in both its simplicity and effectiveness, and its use goes back more than 25 years (see <a class="footnote-reference" href="#i7" id="id5">[4]</a> and <a class="footnote-reference" href="#i8" id="id6">[5]</a> as well as the prosite website at <a class="reference external" href="http://prosite.expasy.org/prosuser.html">http://prosite.expasy.org/prosuser.html</a>).  If the reader has access to a command line (standard in Linux and Mac operating systems and available as Cygwin for Windows), a few simple commands will illustrate the point.:</p>
<div class="highlight-python"><pre>mkdir bacteria
cd bacteria
wget ftp://ftp.ncbi.nih.gov/genomes/Bacteria/all.faa.tar.gz
tar -zxf all.faa.tar.gz
for i in `ls -1 |tr '\n' ' '`;do cat $i/*.faa &gt; $i/all.p.fas;
    tr -d '\r\n' &lt; $i/all.p.fas |tr '&gt;' '\n' &gt; $i/all.1; done
grep GADDY */all.1
grep GG.R.GEME */all.1</pre>
</div>
<p>If wget is not installed on your system, simply download the file by some other means and place it alone in a directory.  In the case of GADDY, tens of thousands of response regulators will be identified.  In the case of GG.R.GEME, you will identify the RNA polymerase in all of the bacterial genomes.  Approximately 20 percent of the proteins containing the 5-letter GADDY motif are not response regulators, but the seven letters of GG.R.GEME has 100 percent specificity.  Given these observation, it is perhaps not surprising that 10 letter motifs (much longer than anything in prosite) have a very high degree of specificity, as indicated when comparing the distribution of BLAST scores of <em>10-mer</em> - matched genes in <em>E. coli</em> and <em>B. subtilus</em>.</p>
<p>Given the high specificity of <em>10-mer</em> peptide exact matches, it is then natural to question the sensitivity of <em>10-mer</em> peptides in searching novel organisms. The right panel of the figure above shows that an average of 80% of the genes in our one-per-genera set of 403 representative bacteria contain a shared amino acid <em>10-mer</em> with another organism.  Both logic and closer inspection of the data reveal that the genes not sharing an amino acid <em>10-mer</em> are not the well-understood proteins, and are not particularly robust choices for assigning phylogeny to a metagenomic read.  Given that modern sequencers are capable of producing 100 million reads in a single sample, it seems prudent to discard the 20% of reads coding for such genes when trying to phylogenetically profile a microbial community.  We are decreasing systematic bias at the cost of a slight increase in counting noise.</p>
<div class="figure">
<img alt="_images/sens_spec_doc.png" src="_images/sens_spec_doc.png" style="width: 100%;" />
<p class="caption">Sensitivity of and specificity of simulated reads from draft soil genomes simulated reads were constructed using MetaSim from genomes of four soil bacteria. <em>Herbaspirillum seropedicae</em> and <em>Bacillus mojavensis</em> are species from genera represented in the BLAST databases NR and NT as well as our signature peptide database (SP). <em>Microbacterium trichotecenolytcum</em> represents a genus found in NR and NT but not in SP. Bosea thiooxidans is from a genus not found in any of the three. <strong>(a)</strong> Specificity of placement of simulated reads on the reference tree using our method. <strong>(b)</strong> Placement of 75-bp reads using our method.  <strong>(c)</strong> Comparison of sensitivity of our method (top right panel) and MEGAN using three different BLAST databases: BLASTX and NR (top left) BLASTN and NT (bottom left), and BLASTX against the same genomes used in SP (bottom right). Simulated read lengths of 75, 150, 300, and 600 bp were used for each of the four genomes in each of the four panels. Colors indicate specificity of placement, with gold indicating non-specific placement near the root node in each case. <strong>(d)</strong> Details of specificity of placement of simulated 150-bp <em>Herbaspirillium seropediacae</em> for the 4 methods: our method (black), MEGAN4 with BLASTX against NR (red), MEGAN4 with BLASTN against NT (green), and MEGAN4 with BLASTX against the same genomes used in SP (cyan).</p>
</div>
<p>Most methods have a higher sensitivity when challenged with data closely related to that in the reference database used to construct the method.  Hence, it is prudent to also check the sensitivity of Sequedex on novel organisms.  In the next figure, we assess the sensitivity of Sequendex on synthetic reads derived from the 99% complete draft genomes of four soil bacteria cultured from a desert soil microbial community.  On the left, we assess the read-length dependence of Sequedex analysis of novel genomic data and compare the results to three established methods.  On the right, we look in more detail at how recruitment to nodes of a phylogenetic tree compares to the more typical method of recruiting to the leaves of a tree, with some post processing.</p>
<p>The relative insensitivity of the phylogenetic profile as the length of genomic fragment increases from 75 base pairs to 300 base pairs is visible in the top panel for the four organisms, and is one of the reasons Sequedex can be used to reliably compare data from different studies and sequencing platforms.  Although the profile does not change with read length, panel (c) shows the sensitivity with which Sequedex identifies reads in comparison with three other methods for each of the four organisms and as a function of read length (75, 150, 300, and 600 base pairs).  We do not attempt to characterize the phylogenetic specificity of assignment in this figure.</p>
<p>To better-understand how the different methods ascribe phylogenetic placement of the synthetic organisms, we present these results for <em>Herbaspirilium seropediacae</em> on a phylogenetic tree in panel (d), on the right side of the figure, averaging over the four read lengths.  Although <em>Herbaspirlium</em> is on the tree (placed on the basis of its RNA polymerase gene), it was not in any of the reference databases.  It is immediately evident that the BLAST-based methods, when run through MEGAN, all place the vast majority of reads overly-specifically, with a small fraction incorrectly assigned to leaves of the tree in the wrong family of bacteria.  Sequedex, by contrast, assigns only about half of the reads to the appropriate node of the phylogenetic tree, where <em>Herbaspirilium</em> joins the phylogenetic tree.  Rather than erroneously assigning the reads, Sequedex chooses to assign them simply as &#8216;bacteria&#8217; or &#8216;beta proteobacteria&#8217;.  While increasing the counting noise in the phylogenetic profile somewhat, this behavior makes the comparison of phylogenetic profiles much less noisy, as the comparisons will be based only on signature peptides which have <em>never</em> been seen outside the specific location on the tree where the assignment is made.</p>
</div>
<div class="section" id="how-can-sequedex-output-be-validated-and-understood">
<h2>2.6. How can Sequedex output be validated and understood?<a class="headerlink" href="#how-can-sequedex-output-be-validated-and-understood" title="Permalink to this headline">¶</a></h2>
<p>Sequedex represents a novel method for classifying genomic data, serving a comparable roll to BLAST and defining protein families with hidden Markov Models.  Although building off of the idea behind such well-established algorithms as PROSITE and BLOCKS, Sequedex leverages modern indexing and text-processing alogrithms, the relatively complete set of reference genomes, and an algorithmic incorporation of inferred phylogeny to produce a qualitative improvement in both the speed and robustness of genomic classification.  Since it represents a novel algorithm, it will require a great deal of empirical application for users to gain a &#8216;feel&#8217; for how to interpret the output of signature based analysis and how it handles the curve balls practical application throws its way.</p>
<p>In later sections, we show the user how to validate with his or her own data. To this end, we see several ways Sequedex can be validated and understood:</p>
<ul class="simple">
<li>Compare results from analysis of files with nucleotide sequences of each gene from completed genomes of various organisms (ffn files)</li>
<li>Compare results from analysis of individual genes from well-annotated organisms</li>
<li>Generate and compare synthetic data</li>
<li>Compare analysis of a variety of microbial communitites</li>
<li>Output, assemble, align, and classify reads.</li>
</ul>
</div>
<div class="section" id="how-will-sequedex-be-improved">
<h2>2.7. How will Sequedex be improved?<a class="headerlink" href="#how-will-sequedex-be-improved" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>More trees (eukaryotes, mitochondria, plastids, archaea, species-level resolution).</li>
<li>Better functional classifications (eukaryotes, improved definitions of functional classifications, better annotation of gene function).</li>
<li>Curation of signature lists to eliminate low complexity signatures, signatures that do not connect orthologs, and mis-called genes.</li>
<li>Provide ability to output peptide fragments on the basis of phylogeny and function, both unaligned and aligned to reference proteins.  This will be useful both for assembly, enzyme mining, and more accurate phylogenetic assignments.</li>
<li>Differentiate deeply branching signatures resulting from conservation and horizontal gene transfer.</li>
<li>Compare metagenome richness and diversity on the basis of metagenomic data, with only weak dependency on the reference database.</li>
<li>Identify and annotate signatures on the basis of metagenomic data, with only weak dependency on the reference database.</li>
</ul>
<p>Sequedex is designed to operate with different signature databases.
At present the only database for public release is the one-per-genus
bacterial database, bact403h.  This database was used to produce the
data in the paper.  We have also have produced databases for viruses,
metazoans, lower eukaryotes, a larger set of bacteria and archaea, and
the complete Tree of Life including viruses.  We do not plan on
releasing these additional databases until after initial publication
of the papers describing them.  Contact the authors if you have unique
access to data that could make use of the broader databases and if you
are interested in collaborating on the initial papers.</p>
<table class="docutils footnote" frame="void" id="i1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td>Berendzen, J., WJ Bruno, JD Cohn, NW Hengartner, CR Kuske, BH McMahon, MA Wolinsky, G Xie, &#8220;Rapid phylogenetic and functional classification of short genomic fragments with signature Peptides&#8221;, BMC Research Notes, August, 2012</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="i2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[2]</a></td><td><a class="reference external" href="http://www.rdmag.com/Awards/Rd-100-Awards/2012/08/DNA-Defined-By-Ancestry/">http://www.rdmag.com/Awards/Rd-100-Awards/2012/08/DNA-Defined-By-Ancestry/</a></td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="i6" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[3]</td><td><em>(<a class="fn-backref" href="#id3">1</a>, <a class="fn-backref" href="#id4">2</a>)</em> <a class="reference external" href="http://www.theseed.org">http://www.theseed.org</a></td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="i7" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id5">[4]</a></td><td>Doolittle R.F. (In) Of URFs and ORFs: a primer on how to analyze derived amino acid sequences., University Science Books, Mill Valley, California, (1986).</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="i8" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[5]</a></td><td>Lesk A.M. (In) Computational Molecular Biology, Lesk A.M., Ed., pp17-26, Oxford University Press, Oxford (1988).</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">2. Introduction to Sequedex</a><ul>
<li><a class="reference internal" href="#what-is-sequedex">2.1. What is Sequedex?</a></li>
<li><a class="reference internal" href="#how-is-sequedex-different-from-other-sequence-analysis-packages">2.2. How is Sequedex different from other sequence analysis packages?</a></li>
<li><a class="reference internal" href="#who-uses-sequedex">2.3. Who uses Sequedex?</a></li>
<li><a class="reference internal" href="#how-does-sequedex-work">2.4. How does Sequedex work?</a></li>
<li><a class="reference internal" href="#what-is-the-specificity-and-sensitivity-of-sequedex-when-matching-bacterial-genes-and-gene-fragments-from-novel-bacteria">2.5. What is the specificity and sensitivity of Sequedex when matching bacterial genes and gene fragments from novel bacteria?</a></li>
<li><a class="reference internal" href="#how-can-sequedex-output-be-validated-and-understood">2.6. How can Sequedex output be validated and understood?</a></li>
<li><a class="reference internal" href="#how-will-sequedex-be-improved">2.7. How will Sequedex be improved?</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="copyright.html"
                        title="previous chapter">1. Copyright notice</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="installation.html"
                        title="next chapter">3. Installation instructions</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/introduction.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="installation.html" title="3. Installation instructions"
             >next</a> |</li>
        <li class="right" >
          <a href="copyright.html" title="1. Copyright notice"
             >previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; <a href="copyright.html">Copyright</a> 2012, Los Alamos National Security, LLC.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>