

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Phylogenetic and functional profiles &mdash; Sequedex 0.9.1 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="Sequedex 0.9.1 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">Sequedex 0.9.1 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="phylogenetic-and-functional-profiles">
<h1>Phylogenetic and functional profiles<a class="headerlink" href="#phylogenetic-and-functional-profiles" title="Permalink to this headline">¶</a></h1>
<p>The ability of Sequedex to profile bacterial communities with metagenomic data leads naturally to the question of how a user&#8217;s microbial community compares to previously sequenced microbial communities.  In this chapter, we lead the user through the process of computing phylogenetic profiles and comparing them to publically available datasets. In additional to phylogenetic information, the sequences obtained in a shotgun metagenomics dataset can be used to obtain functional profiles of genes in a microbial community, identify the presence of particular genes of interest, or even determine which of proteins are the most important determinants of classification within a set of microbial communities.  In this chapter, we we lead the user through the process of computing functional profiles.</p>
<div class="section" id="acquiring-data-files">
<h2>Acquiring data files:<a class="headerlink" href="#acquiring-data-files" title="Permalink to this headline">¶</a></h2>
<p>Metagenomics data can be acquired several sources, including the Sequence Read Archive at NCBI, where the production phase of the human microbiome project can be found with the project number, SRP002163 and the study number SRP002163.  Individual data sets can be downloaded from their ftp site with a web browser (or using wget) at <a class="reference external" href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/litesra/SRP/SRP002163/SRR059366">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/litesra/SRP/SRP002163/SRR059366</a>.  Metadata about data sets, such as it is, can be obtained with the Run Browser at the Sequence Read Archive, <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser/">http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser/</a>.</p>
<p>Environmental metagenomics data sets can be found at NCBI with the taxonid 410658.  A wide variety of individual projects can be found, including samples from Pru Toh Daeng Peat Swamp, in Southern Thailand: SRR023820, which can be downloaded from <a class="reference external" href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP001/SRP001114/SRR023820/SRR023820.sra">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP001/SRP001114/SRR023820/SRR023820.sra</a>.</p>
<p>In order to process data sets from the Sequence Read Archive, it will be necessary to extract fasta files from the .sra files obtained above.  This can be done with the fastq-dump utility from the SRA Toolkit, available for Linux, Mac, and Windows platforms at <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</a>.  Generation of a fasta file from the HMB dataset above requires the command fastq-dump &#8211;fasta SRR059366.sra.</p>
<p>The Department of Energy&#8217;s Joint Genome institute also maintains a repository of metagenomic sequence data at their integrated metagenomics (IM) site.</p>
<p>The CREO data from reference [1] can be downloaded in fasta format at <a class="reference external" href="http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=MetaDetail&amp;downloadTaxonReadsFnaFile=1&amp;taxon_oid=2029527002&amp;noHeader=1">http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=MetaDetail&amp;downloadTaxonReadsFnaFile=1&amp;taxon_oid=2029527002&amp;noHeader=1</a>, while additional metagenomics datasets are listed at <a class="reference external" href="http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonList&amp;page=taxonListAlpha&amp;domain=*Microbiome">http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonList&amp;page=taxonListAlpha&amp;domain=*Microbiome</a>.</p>
<p>The Community Cyberinfrastructure for Advanced Microbial Ecology Research and Analysis site (CAMERA) also maintains a repository of metagenomics sequence data at <a class="reference external" href="https://portal.camera.calit2.net/gridsphere/gridsphere">https://portal.camera.calit2.net/gridsphere/gridsphere</a>, where we were able to download data from the Global Ocean Survey (DB Rusch, et al. &#8220;The SorcererII Global Ocean Sampleing Expedition: Northwest Atlantic through Eastern Tropical Pacific&#8221; PLoS Biology, March 2007).</p>
<p>Using your own data requires only that it be put into the common fasta file format.  Since most sequencers supply data in fastq format (which supplies base-by-base quality scores in addition to to the amino acid sequence), you may need to convert the data from fastq to fasta.  Numerous utilities are available to do this, but a particularly simply way is available from teh Edwards lab at San Diego State University at <a class="reference external" href="http://edwards.sdsu.edu/labsite/index.php/robert/289-how-to-convert-fastq-to-fasta">http://edwards.sdsu.edu/labsite/index.php/robert/289-how-to-convert-fastq-to-fasta</a>, with the nearly unreadable one-line perl script::</p>
<div class="highlight-python"><pre>cat file_in.fastq | perl -e '$i=0;while(&lt;&gt;){if(/^\@/&amp;&amp;$i==0){s/^\@/\&gt;/;print;}elsif($i==1){print;$i=-3}$i++;}' &gt; file_out.fasta.</pre>
</div>
</div>
<div class="section" id="running-sequedex">
<h2>running Sequedex<a class="headerlink" href="#running-sequedex" title="Permalink to this headline">¶</a></h2>
<p>However you acquire your data, the fasta files to be analyzed by Sequedex in a given batch should all be placed in a single directory.  The analysis is run as described for your computing platform in the &#8216;getting started&#8217; portion of this user&#8217;s manual. For example, by typing &#8220;sequescan bact403.1.a <a href="#id1"><span class="problematic" id="id2">*</span></a>.fasta ./b403/&#8221;.  This will create a directory b403 where output files corresponding to each input file will be placed.  For this section, we will concern ourselves with the phylogenetic profiles, that appear in the files &#8216;infile.nodes&#8217;, corresponding to each input fasta file.  For the next steps, we will show the results for two sets of fasta files: a set of 258 environmental metagenomes, and a set of 400 metagenomes from the human microbiome project.</p>
</div>
<div class="section" id="comparison-of-phylogenetic-and-functional-profiles">
<h2>comparison of phylogenetic and functional profiles<a class="headerlink" href="#comparison-of-phylogenetic-and-functional-profiles" title="Permalink to this headline">¶</a></h2>
<p>For those users wanting to look at output data as rapidly as possible, we supply two Excel spreadsheets with the collected results for the human microbiome data (hmbm.xlsx) and environmental metagenomes (env4.xlsx), and he or she can skip directly to the next section.</p>
<p>When comparing phylogenetic profiles of metagenomes, it is helpful to produce a matrix of phylogenetic profiles, with the 403 rows corresponding to the nodes on the bacterial phylogeny, and each sample assigned its own column, as well as a matrix of functional profiles, with the 982 rows corresponding to the SEED (<a class="reference external" href="http://www.theseed.org">http://www.theseed.org</a>) functional categories.  This can be done in many ways, but a simple shell-script will do the job::</p>
<div class="highlight-python"><pre>cd b403
for i in SRR059330 SRR059331 SRR059338 SRR059339; do
awk '{print $3}' &lt; $i.fasta.nodes &gt; $i.j;
awk '{print $3}' &lt; $i.fasta.fun &gt; $i.k;
done
paste SRR059330.j SRR059331.j SRR059338.j SRR059339.j &gt; ../hmb.b403.ph
paste SRR059330.k SRR059331.k SRR059338.k SRR059339.k &gt; ../hmb.b403.f</pre>
</div>
<p>will combine the output files from four stool metagenomics data sets into a matrix of counts.</p>
<p>From these raw counts files, several types of profiles can easily be computed: phylogenetic rollups, normalized phylogenetic profiles, and normalized scalar products of the profiles.  Once again, these quantities can be computed in many ways, but the gfortran (<a class="reference external" href="http:/gcc.gnu.org/wiki/GFortran/">http:/gcc.gnu.org/wiki/GFortran/</a>) code pnorm.f90 and fnorm.f90 will suffice.  In particular, examination of the tree (bact403.phyloxml) with archyoptrix (<a class="reference external" href="http://www.phylosoft.org">http://www.phylosoft.org</a>) reveals the following roll-up categories::</p>
<div class="highlight-python"><pre>ru(i,1)=sum(k(i,5:19))  !cyanobacteria
ru(i,2)=sum(k(i,20:69))+sum(k(i,72:79)) !firmicutes
ru(i,3)=sum(k(i,82:88)) !chloroflexi
ru(i,4)=sum(k(i,90:92)) !synergistetes
ru(i,5)=sum(k(i,94:95)) ! Deinococcus
ru(i,6)=sum(k(i,97:99)) ! thermotogae
ru(i,7)=sum(k(i,100:149)) !actinobacteria
ru(i,8)=sum(k(i,151:248)) !b/g proteobacteria
ru(i,9)=sum(k(i,255:256))  !acidobacteria
ru(i,10)=sum(k(i,258:261))  !aquificae
ru(i,11)=sum(k(i,265:291)) !delta / epsilon proteobacteria
ru(i,12)=sum(k(i,293:350)) !alpha proteobacteria
ru(i,13)=sum(k(i,352:355)) !spirochaetes
ru(i,14)=sum(k(i,358:361)) !planctomycetes
ru(i,15)=sum(k(i,362:368)) !chlamydiae+verrucomicrobia
ru(i,16)=sum(k(i,370:402)) !bacteroidetes
ru(i,17)=k(i,2)+k(i,3)+k(i,4)+k(i,70)+k(i,71)+k(i,80)+k(i,81)+k(i,89)+k(i,93)+k(i,96)+k(i,150)+sum(k(i,249:254))+k(i,257)+k(i,292)+k(i,351)+sum(k(i,356:357))+k(i,369) !multiphyla
ru(i,18)=k(i,1) !root</pre>
</div>
<div class="figure">
<img alt="hmb_proll.*" src="hmb_proll.*" />
<p class="caption">Phylogenetic profile of 400 human microbiome samples, computed using the bact403 tree and Sequedex, rolled up with the above-defined classifications.</p>
</div>
<p>similarly, inspection of the output file SRR059330.fasta.fun provides the following rollup categories for function::</p>
<div class="highlight-python"><pre>ru(i,1)=sum(k(i,1:54))  !amino acids and derivatives
ru(i,2)=sum(k(i,55:153)) !carbohydrates
ru(i,3)=sum(k(i,154:162)) !cell division and cell cycle
ru(i,4)=sum(k(i,163:213)) !cell wall and capsule
ru(i,5)=sum(k(i,214:357)) ! clustering-based
ru(i,6)=sum(k(i,358:397)) !cofactors, vitamins, prosthetic groups, pigments
ru(i,7)=sum(k(i,398:431)) !dna metabolism
ru(i,8)=sum(k(i,432:445)) !dormancy and sporulation
ru(i,9)=sum(k(i,446:464))  !fatty acids, lipids, and isoprenoids
ru(i,10)=sum(k(i,465:488))  !Iron acquisition and metabolism
ru(i,11)=sum(k(i,489:534)) !membrane transport
ru(i,12)=sum(k(i,535:564)) !metablism of aeromatic compounds
ru(i,13)=sum(k(i,565:579)) !miscellaneous
ru(i,14)=sum(k(i,580:587)) !motility and chemotaxis
ru(i,15)=sum(k(i,588:597)) !nitrogen metabolism
ru(i,16)=sum(k(i,598:615)) !nucleosides and nucleotides
ru(i,17)=sum(k(i,616:637)) !phages, prophages, transposable elements
ru(i,18)=sum(k(i,638:643)) !phosphorus metabolism
ru(i,19)=sum(k(i,644:652)) !photosynthesis
ru(i,20)=sum(k(i,653:656)) !potassium metabolism
ru(i,21)=sum(k(i,657:722)) !protein metabolism
ru(i,22)=sum(k(i,723:753)) !RNA metabolism
ru(i,23)=sum(k(i,754:791)) !regulation and cell signaling
ru(i,24)=sum(k(i,792:827)) !respiration
ru(i,25)=sum(k(i,828:843)) !secondary metabolism
ru(i,26)=sum(k(i,844:881)) !stress response
ru(i,27)=sum(k(i,882:894)) !sulfur metabolism
ru(i,28)=sum(k(i,895:962)) !virulence
ru(i,29)=k(i,963) !no match</pre>
</div>
</div>
<div class="section" id="comparison-of-metagenomes">
<h2>comparison of metagenomes<a class="headerlink" href="#comparison-of-metagenomes" title="Permalink to this headline">¶</a></h2>
<p>The set of output files can be viewed in Excel, and we supply two Excel spreadsheets with the collected results for the human microbiome data <tt class="xref download docutils literal"><span class="pre">hmb.xlsx</span></tt> and environmental metagenomes (env4.xlsx).  In the next section we will consider several quantitative analyses.</p>
</div>
<div class="section" id="analyzing-output-with-r">
<h2>Analyzing output with R<a class="headerlink" href="#analyzing-output-with-r" title="Permalink to this headline">¶</a></h2>
<p>R is a powerful open source software for statistical data analysis.
Binaries and download instruction are found at <a class="reference external" href="http://cran.r-project.org/">http://cran.r-project.org/</a>.
In addition to the basic installation, many users will want to peruse the
library of contributed packages that extend the functionality and
methods available.  Packages to consider include ape,</p>
</div>
<div class="section" id="clustering-with-r">
<h2>clustering with R<a class="headerlink" href="#clustering-with-r" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="pairwise-comparisons-with-r">
<h2>pairwise comparisons with R<a class="headerlink" href="#pairwise-comparisons-with-r" title="Permalink to this headline">¶</a></h2>
<p>R has many useful add-on libraries available.  The ape library provides functions
for reading and manipulating phylogenetic trees and DNA sequences,
computing DNA distances, estimating trees with distance-based methods, and a
range of methods for comparative analyses and analysis of diversification.
Instruction on how to install libraries are found at the CRAN
<a class="reference external" href="http://cran.r-project.org/">http://cran.r-project.org/</a>.  Once installed, it is possible to display
the results on the analysis on phylogenetic trees. In this analysis, we show
how one can compare two metagenomic samples by placing on a phylogenetic tree
significant differences.  The tree we use in our example is in newick format. ::</p>
<div class="highlight-python"><div class="highlight"><pre><span class="c">#  Load the library ape</span>
</pre></div>
</div>
<p>library(ape)</p>
<p>#  read the phylogenetic tree in newick format of 403 representative genomes
rna.tree      &lt;- read.tree(file=&#8221;Bacteria403.newwick&#8221;)</p>
<p>#  identify the internal nodes and the tips
nodeDepth     &lt;- node.depth(rna.tree)
index.node    &lt;- 1:length(nodeDepth)
node.internal &lt;- index.node[nodeDepth &gt; 1]
index.tip     &lt;- index.node[nodeDepth==1]</p>
<p>#  simplify names at tip and keep only 1 in 5 of them
tipname.tmp   &lt;- rna.tree$tip.label
tipname       &lt;- sapply(strsplit(tipname.tmp,&#8221;_&#8221;),</p>
<blockquote>
<div>FUN=function(x) x[1])</div></blockquote>
<p>rna.tree$tip.label &lt;- paste(&#8221;     &#8221;,tipname,&#8221;     &#8221;)
rna.tree$tip.label[index.tip %% 5 != 1]  &lt;- &#8220;&#8221;</p>
<p>#  read a dataset from the face experiment from a tab-delimited file
face &lt;- read.table(file=&#8221;face_node_counts.txt&#8221;,</p>
<blockquote>
<div>header=T,
sep=&#8221;t&#8221;)</div></blockquote>
</div>
<div class="section" id="niche-determinants-gulf-oil">
<h2>niche determinants, gulf oil<a class="headerlink" href="#niche-determinants-gulf-oil" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="tooth-decay-study">
<h2>tooth decay study<a class="headerlink" href="#tooth-decay-study" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Phylogenetic and functional profiles</a><ul>
<li><a class="reference internal" href="#acquiring-data-files">Acquiring data files:</a></li>
<li><a class="reference internal" href="#running-sequedex">running Sequedex</a></li>
<li><a class="reference internal" href="#comparison-of-phylogenetic-and-functional-profiles">comparison of phylogenetic and functional profiles</a></li>
<li><a class="reference internal" href="#comparison-of-metagenomes">comparison of metagenomes</a></li>
<li><a class="reference internal" href="#analyzing-output-with-r">Analyzing output with R</a></li>
<li><a class="reference internal" href="#clustering-with-r">clustering with R</a></li>
<li><a class="reference internal" href="#pairwise-comparisons-with-r">pairwise comparisons with R</a></li>
<li><a class="reference internal" href="#niche-determinants-gulf-oil">niche determinants, gulf oil</a></li>
<li><a class="reference internal" href="#tooth-decay-study">tooth decay study</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/phylogenetic.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">Sequedex 0.9.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; <a href="copyright.html">Copyright</a> 2012, Los Alamos National Security, LLC.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>