

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>5. Phylogenetic and functional profiles &mdash; Sequedex 0.9.2 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/tree.ico"/>
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="Sequedex 0.9.2 documentation" href="index.html" />
    <link rel="next" title="6. Tree of Life" href="tree_of_life.html" />
    <link rel="prev" title="4. Getting started" href="starting.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="tree_of_life.html" title="6. Tree of Life"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="starting.html" title="4. Getting started"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="phylogenetic-and-functional-profiles">
<h1>5. Phylogenetic and functional profiles<a class="headerlink" href="#phylogenetic-and-functional-profiles" title="Permalink to this headline">¶</a></h1>
<p>It is assumed at this point that the user is comfortable running Sequescan and can obtain multiple output files for comparison.  Please refer to the previous two chapters for installation, platform-dependent instructions on running Sequescan with both the GUI and the command line, and instructions on obtaining a license.</p>
<p>The ability of Sequedex to profile bacterial communities with metagenomic data leads naturally to the question of how a user&#8217;s microbial community compares to previously sequenced microbial communities.  In this chapter, we lead the user through the process of computing phylogenetic profiles and comparing them to publically available datasets. In additional to phylogenetic information, the sequences obtained in a shotgun metagenomics dataset can be used to obtain functional profiles of genes in a microbial community, identify the presence of particular genes of interest, or even determine which proteins are the most important determinants of classification within a set of microbial communities.  In this chapter, we we lead the user through the process of computing functional profiles.</p>
<p>A note of caution is warranted here.  In this and the next section we present phylogenetic and functional profiles, as well as analysis of these profiles across an extraordinary range of microbial communities with samples prepared by different researchers with a variety of purposes, sequenced by different methods with different read-lengths.  We have primarily relied on publically available datasets with associated publications, although a few examples of unpublished work with experimental collaborators is also included.  Our purpose in this user&#8217;s manual is to provide illustrative examples of how phylogenetic and functional profiles can be visualized and analyzed for both self-consistency and biological insights, and provide analysis of a reference collection of data to which the user can compare his or her own data.  The job of cross-checking these insights with further experimentation analysis is distinct and belongs in the peer-reviewed literature.  Our intent here is to put this analysis tool into the hands of the people best-suited to perform these cross-checks - those who decide which samples will be prepared and know what biological phenomonena are expected.</p>
<div class="section" id="acquiring-data-files">
<h2>5.1. Acquiring data files<a class="headerlink" href="#acquiring-data-files" title="Permalink to this headline">¶</a></h2>
<p>Sequedex at present requires input sequence data files be in the fasta data format.  Sequencing platforms typically output sequences in a format with quality scores, such as fastq, which can be used as input to assemblers.  Because possibility space of 10-mers of amino acids is so much larger than the number of signatures, Sequedex is relatively insensitive to sequencing errors and the quality scores are ignored, and the fasta file format is suitable.  Because fasta files have less information than the output of sequencers, it is relatively simple to change most formats into fasta format.  While preprocessing of reads to remove duplicates is valuable, and minimal quality score filtering is probably useful, assembling the data into contigs before profiling the community is probably unhelpful, as it makes the profiles dependent on the depth of sequencing.</p>
<p>For users wishing to compare their data to others, metagenomics data can be acquired from several sources, including the Sequence Read Archive at NCBI, where the production phase of the human microbiome project can be found with the project number, SRP002163 and the study number SRP002163.  Individual data sets can be downloaded from their ftp site with a web browser (or using wget) at <a class="reference external" href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR059/SRR059366/SRR059366.sra">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR059/SRR059366/SRR059366.sra</a>.  Metadata about data sets, such as it is, can be obtained with the Run Browser at the Sequence Read Archive, <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser/">http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser/</a>.</p>
<p>Environmental metagenomics data sets can be found at NCBI with the taxonid 410658.  A wide variety of individual projects can be found, including samples from Pru Toh Daeng Peat Swamp, in Southern Thailand: SRR023820, which can be downloaded from <a class="reference external" href="ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP001/SRP001114/SRR023820/SRR023820.sra">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP001/SRP001114/SRR023820/SRR023820.sra</a>.</p>
<p>In order to process data sets from the Sequence Read Archive, it will be necessary to extract fasta files from the .sra files obtained above.  This can be done with the fastq-dump utility from the SRA Toolkit, available for Linux, Mac, and Windows platforms at <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software</a>.  Generation of a fasta file from the HMB dataset above requires the command:</p>
<div class="highlight-python"><pre>fastq-dump --fasta SRR059366.sra</pre>
</div>
<p>The Department of Energy&#8217;s Joint Genome institute also maintains a repository of metagenomic sequence data at their integrated metagenomics (IM) site.</p>
<p>The CREO data from reference [1] can be downloaded in fasta format at <a class="reference external" href="http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=MetaDetail&amp;downloadTaxonReadsFnaFile=1&amp;taxon_oid=2029527002&amp;noHeader=1">http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=MetaDetail&amp;downloadTaxonReadsFnaFile=1&amp;taxon_oid=2029527002&amp;noHeader=1</a>, while additional metagenomics datasets are listed at <a class="reference external" href="http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonList&amp;page=taxonListAlpha&amp;domain=*Microbiome">http://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonList&amp;page=taxonListAlpha&amp;domain=*Microbiome</a>.</p>
<p>The Community Cyberinfrastructure for Advanced Microbial Ecology Research and Analysis site (CAMERA) also maintains a repository of metagenomics sequence data at <a class="reference external" href="https://portal.camera.calit2.net/gridsphere/gridsphere">https://portal.camera.calit2.net/gridsphere/gridsphere</a>, where we were able to download data from the Global Ocean Survey <a href="#id3"><span class="problematic" id="id4"><span id="id1"></span>[#f1]_</span></a>.</p>
<p>Using your own data requires only that it be put into the common fasta file format.  Since most sequencers supply data in fastq format (which supplies base-by-base quality scores in addition to to the amino acid sequence), you may need to convert the data from fastq to fasta.  Numerous utilities are available to do this, but a particularly simply way is available from the Edwards lab at San Diego State University at <a class="reference external" href="http://edwards.sdsu.edu/labsite/index.php/robert/289-how-to-convert-fastq-to-fasta">http://edwards.sdsu.edu/labsite/index.php/robert/289-how-to-convert-fastq-to-fasta</a>, with this one-line perl script::</p>
<div class="highlight-python"><pre>cat file_in.fastq | perl -e '$i=0;while(&lt;&gt;){if(/^\@/&amp;&amp;$i==0){s/^\@/\&gt;/;print;
}elsif($i==1){print;$i=-3}$i++;}' &gt; file_out.fasta.</pre>
</div>
</div>
<div class="section" id="phylogenetic-rollups">
<h2>5.2. Phylogenetic rollups<a class="headerlink" href="#phylogenetic-rollups" title="Permalink to this headline">¶</a></h2>
<p>For users wanting to look at output data as rapidly as possible, we supply two Excel spreadsheets with the collected results for the <tt class="xref download docutils literal"><span class="pre">human</span> <span class="pre">microbiome</span> <span class="pre">data</span></tt> and <tt class="xref download docutils literal"><span class="pre">environmental</span> <span class="pre">metagenomes</span></tt>, and he or she can skip directly to the next section.  The human microbiome data was obtained from NCBI&#8217;s sequence read archive, and the individual runs analyzed can be obtained by pasting the run number at the top of each column of the above spreadsheet into their run browser, at <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser">http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser</a>.</p>
<p>Although this data set was assembled prior to publication, a description of the project aims and primary analysis results can be found in &#8220;A framework for human microbiome research&#8221; by the Human Microbiome Consortium, in <a class="reference external" href="http://www.nature.com/nature/journal/v486/n7402/full/nature11209.html">http://www.nature.com/nature/journal/v486/n7402/full/nature11209.html</a>.</p>
<p>When comparing phylogenetic profiles of metagenomes, it is helpful to produce a matrix of phylogenetic profiles, with the 403 rows corresponding to the nodes on the bacterial phylogeny, and each sample assigned its own column, as well as a matrix of functional profiles, with the 982 rows corresponding to the SEED (<a class="reference external" href="http://www.theseed.org">http://www.theseed.org</a>) functional categories.  This can be done in many ways, including reading each output file into Excel and pasting the relevant columns into one sheet, but a simple shell-script will also do the job.</p>
<div class="highlight-python"><pre>cd output/sequescan-0.9-infile/
for i in SRR059330 SRR059331 SRR059338 SRR059339; do
awk '{print $4}' &lt; $i.fasta/$i.fasta-bact403.1-phylogeny_counts.tsv &gt; $i.j;
awk '{print $2}' &lt; $i.fasta/$i.fasta-bact403.1-seed_0911-function.counts.tsv &gt; $i.k;
done
paste SRR059330.j SRR059331.j SRR059338.j SRR059339.j &gt; ../hmb.b403.ph
paste SRR059330.k SRR059331.k SRR059338.k SRR059339.k &gt; ../hmb.b403.f</pre>
</div>
<p>will combine the output files from four stool metagenomics data sets into a matrix of counts.  You will, of course, need to supply directory and file names corresponding to your particular analysis.  If the</p>
<p>From these raw counts files, several types of profiles can easily be computed: phylogenetic rollups, normalized phylogenetic profiles, and normalized scalar products of the profiles.  Once again, these quantities can be computed in many ways, but the gfortran (<a class="reference external" href="http:/gcc.gnu.org/wiki/GFortran/">http:/gcc.gnu.org/wiki/GFortran/</a>) <tt class="xref download docutils literal"><span class="pre">code</span></tt> will suffice.  In particular, examination of the <a class="reference download internal" href="_downloads/Bacteria4031.phyloxml"><tt class="xref download docutils literal"><span class="pre">tree</span></tt></a> with archyoptrix (<a class="reference external" href="http://www.phylosoft.org">http://www.phylosoft.org</a>) or an <a class="reference download internal" href="_downloads/infile.fasta.nodes"><tt class="xref download docutils literal"><span class="pre">output</span> <span class="pre">file</span></tt></a> reveals ninteen roll-up categories, defined in the notation of Fortran 90 <tt class="xref download docutils literal"><span class="pre">here</span></tt>.  Only eight of the most highly represented categories are shown in the human microbiome plot, while the environmental metagenomes rollup, below, shows seventeen categories, with only &#8216;multiphyla&#8217; and &#8216;root&#8217; left off.</p>
<div class="figure">
<img alt="_images/hmbm_proll.png" src="_images/hmbm_proll.png" style="width: 100%;" />
<p class="caption">Phylogenetic profile of 400 human microbiome samples, computed using the bact403 tree and Sequedex, rolled up with the above-defined classifications.  The original plot and data can be found in the first panel of <tt class="xref download docutils literal"><span class="pre">this</span> <span class="pre">Excel</span> <span class="pre">spreadsheet</span></tt>.  For the most parts, the samples consist of pairs of replicates, and they are nearly indistinguishable from each other in this presentation.</p>
</div>
<p>The phylogenetic and functional profiles presented here can be compared to Figure 2 of &#8220;Structure, function, and diversity of the health human microbiome&#8221;, also by the Human Microbiome Consortium, in <a class="reference external" href="http://www.nature.com/nature/journal/v486/n7402/full/nature11234.html">http://www.nature.com/nature/journal/v486/n7402/full/nature11234.html</a>.  Although the Human Microbiome Consortium ran their datasets through BLASTX (see SRS016585 at <a class="reference external" href="http://www.hmpdacc.org/HMSCP/#data">http://www.hmpdacc.org/HMSCP/#data</a> to verify that stool sample run SRR059346 is indeed primarily <em>E. coli</em>), the analysis in the HMB paper utilized a rapid-matching scheme, described in Segata, et al., &#8220;Metagenomic microbial community profiling using unique clade-specific marker genes&#8221; Nature Methods 9:811-814 (2012) <a class="reference external" href="http://www.nature.com/nmeth/journal/v9/n8/full/nmeth.2066.html">http://www.nature.com/nmeth/journal/v9/n8/full/nmeth.2066.html</a>.</p>
<p>Although Segata, et al. used clade-specific genes to profile metagenomics data, while we used phylogenetic signatures of the majority of genes, numerous points of agreement between the analyses are evident.  Note that we included a &#8216;multiphyla&#8217; category in our figure, which was not included in their reported results.  Specifically:</p>
<ul class="simple">
<li>For the stool samples, the ratio of bacteroidetes to clostridia varies from 60% clostridia to 95% bacteroidetes.</li>
<li>For the tongue dosrum samples, firmicutes range from half to 10% of the identified species, with proteobacteria next most abundant, followed by bacteroidetes.</li>
<li>For the buccal mucosa samples, firmicutes are even more prevalent, together with proteobacteria making up more than 90% of the assigned reads.</li>
<li>For the supragingival plaque samples, roughly equal representation of firmicutes, actinobacteria, proteobacteria, and bacteroidetes occurs, with considerable variability in the relative abundance.</li>
<li>For the anterior nares samples, actinobacteria makes up more than 90% of the assigned reads in some samples, with the firmicutes providing the bulk of the balance.</li>
<li>For the posterior fornix samples, most of the samples consist of firmicutes (clostridia + bacilli), although three of the analyzed by Sequedex consisted of equal parts bacteroidetes and actinobacteria, as did one in Figure 2 of the HMB paper.</li>
<li>For the retroauricular crease samples, most of the samples are almost entirely actinobacteria, while alpha proteobacteria make up a significant minority of several of the samples.</li>
<li>The samples labeled &#8216;mouse&#8217; were obtained from William Hsiao of the British Columbia CDC and University of British Columbia, in a controlled comparison of mouse stool samples of mice with and without receiving an antibiotic cocktail.  Although we were unable to find the relevant metadata, it appears that the clonal population of <em>E. coli</em> observed in SRR059346 likely resulted from a patient receiving antibiotics.</li>
</ul>
<p>We will return to further analysis of the HMB dataset after examining a set of representative environmental microbiomes.</p>
<div class="figure">
<img alt="_images/env4_proll.png" src="_images/env4_proll.png" style="width: 100%;" />
<p class="caption">Phylogenetic profile of 242 environmental microbiome samples,  computed using the bact403 tree and Sequedex, rolled up with the above-defined classifications.  The original plot and data can be found in the first panel of <tt class="xref download docutils literal"><span class="pre">this</span> <span class="pre">Excel</span> <span class="pre">spreadsheet</span></tt>.  The data are a compilation of 27 distinct studies, with citations provided below, in order from left to right on the figure.  The data come primarily from the sequence read archive (indicated by the SRR number above the relevent columns in the spreadsheet.  The Global Ocean Survey results can be obtained from CAMERA (<a class="reference external" href="https://portal.camera.calit2.net/">https://portal.camera.calit2.net/</a>) and the FACE data from JGI/IMG (<a class="reference external" href="http://img.jgi.doe.gov/">http://img.jgi.doe.gov/</a>).</p>
</div>
<p>A cursory examination of the environmental microbiome profiles reveals distinctive and repeatable differences between studies.  We present here references to the publications from each sample, and a brief summary of salient details.</p>
<p><strong>peat</strong> Kanokratana, et al., &#8220;Insights into the phylogeny and metabolic potential of a primary tropical peat swamp forest microbial community by metagenomic analysis&#8221; <em>Microb. Ecol.</em> <strong>61</strong>:518-528 (2011). (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21057783">http://www.ncbi.nlm.nih.gov/pubmed/21057783</a>)</p>
<p><strong>Permafrost1</strong> Yergeau E, Hogues H, Whyte LG, Greer CW. &#8220;The functional potential of high Arctic permafrost revealed by metagenomic sequencing, qPCR and microarray analyses.&#8221; ISME J. 2010 Sep;4(9):1206-14. (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20393573">http://www.ncbi.nlm.nih.gov/pubmed/20393573</a>)</p>
<p><strong>arctic</strong> Yergeau E, Sanschagrin S, Beaumier D, Greer CW, &#8220;Metagenomic Analysis of the Bioremediation of Diesel-Contaminated Canadian High Arctic Soils. PLoS ONE <strong>7</strong>:e30058 (2012) (<a class="reference external" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0030058">http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0030058</a>)</p>
<p><strong>Nyegga</strong> Stokke R, Roalkvam I, Lanzen A, Haflidason H, Steen IH., &#8220;Integrated metagenomic and metaproteomic analyses of an ANME-1-dominated community in marine cold seep sediments&#8221; Environ Microbiol. 2012 May;14(5):1333-1346 (2012) (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22404914">http://www.ncbi.nlm.nih.gov/pubmed/22404914</a>).  Anaerobic methanotrophic archaea (ANME) and sulfur metabolising bacteria presumably include the delta proteobacteria visible in this sample.</p>
<p><strong>Harvard</strong> FJ Stewart, AK Sharma, JA Bryant, JM Eppley, and EF DeLong &#8220;Community transcriptomics reveals universal patterns of protein sequence conservation in natural microbial communities&#8221; Genome Biology <strong>12</strong>:R26 (2011). (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3129676/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3129676/</a>)</p>
<p><strong>permafrost2</strong> Mackelprang R, Waldrop MP, DeAngelis KM, David MM, Chavarria KL, Blazewicz SJ, Rubin EM, Jansson JK. &#8220;Metagenomic analysis of a permafrost microbial community reveals a rapid response to thaw.&#8221; Nature. 2011 Nov 6;480(7377):368-71. (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22056985">http://www.ncbi.nlm.nih.gov/pubmed/22056985</a>)</p>
<p><strong>gulf2</strong> Widger, et al., &#8220;Longitudinal Metagenomic Analysis of the Water and Soil from Gulf of Mexico Beaches Affected by the Deep Water Horizon Oil Spill&#8221; Preceedings of Nature <a class="reference external" href="hdl:10101/npre.2011.5733.1">hdl:10101/npre.2011.5733.1</a> (<a class="reference external" href="http://precedings.nature.com/documents/5733/version/1">http://precedings.nature.com/documents/5733/version/1</a>).  The first eight samples are from sand, the second eight are from water.</p>
<p><strong>face</strong> Berendzen, et al, &#8220;Rapid phylogenetic and functional classification of short genomic fragments with signature peptides&#8221; BMC Research Notes <strong>5</strong>:460 (2012). (<a class="reference external" href="http://www.biomedcentral.com/1756-0500/5/460/abstract">http://www.biomedcentral.com/1756-0500/5/460/abstract</a>).  For a description of the FACE sites, see <a class="reference external" href="http://public.ornl.gov/face/results.shtml">http://public.ornl.gov/face/results.shtml</a>.</p>
<p><strong>utah</strong> Kuske, CR, et al., Soil crusts in Utah field site.  manuscript in preparation.</p>
<p><strong>MOOM</strong> Canfield, DE, et al., &#8220;A cryptic sulfur cycle in oxygen-minimum-zone waters off the Chilean coast&#8221;  Science <strong>330</strong>:1375 (2010).</p>
<p><strong>E. channel</strong> Gilbert, JA, et al., &#8220;Metagenomes and metatranscriptomes from the L4 long-term coastal monitoring station in the Western English Channel&#8221; Standards in Genomic Sciences <strong>3</strong>:183-193 (2010).  Only functional analysis is presented, with SEED and MG-RAST.</p>
<p><strong>carbon</strong> Mou, X., S. Sun, RA Edwards, RE Hodson, MA Moran &#8220;Bacterial carbon processing by generalist species in the coastal ocean&#8221; Nature <strong>451</strong>:708-711 (2008). (<a class="reference external" href="http://www.nature.com/nature/journal/v451/n7179/abs/nature06513.html">http://www.nature.com/nature/journal/v451/n7179/abs/nature06513.html</a>)</p>
<p><strong>acid</strong> Gilbert JA, Thomas S, Cooley NA, Kulakova A, Field D, Booth T, McGrath JW, Quinn JP, Joint I., &#8220;Potential for phosphonoacetate utilization by marine bacteria in temperate coastal waters.&#8221; Environ Microbiol. <strong>11</strong>:111-125 (2009) (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/18783384">http://www.ncbi.nlm.nih.gov/pubmed/18783384</a>)</p>
<p><strong>aloha</strong> Martinez A, Tyson GW, Delong EF., &#8220;Widespread known and novel phosphonate utilization pathways in marine bacteria revealed by functional screening and metagenomic analyses.&#8221; Environ Microbiol. <strong>12</strong>:222-238 (2010) (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19788654">http://www.ncbi.nlm.nih.gov/pubmed/19788654</a>)</p>
<p><strong>Ace Lake</strong> Lauro, FM, et al., &#8220;An integrative study of a meromictic lake exosystem in Antarctica&#8221; ISME J. <strong>5</strong>:879-895 (2011) (<a class="reference external" href="http://www.nature.com/ismej/journal/v5/n5/abs/ismej2010185a.html">http://www.nature.com/ismej/journal/v5/n5/abs/ismej2010185a.html</a>)</p>
<p><strong>Organic Lake</strong> Yau, S., et al. &#8220;Virophage control of antarctic algal host–virus dynamics&#8221; PNAS, USA <strong>108</strong>:6163-6168 (2011). (<a class="reference external" href="http://www.pnas.org/content/early/2011/03/24/1018221108.full.pdf+html">http://www.pnas.org/content/early/2011/03/24/1018221108.full.pdf+html</a>) Primarily about viruses.</p>
<p><strong>stromatolites</strong> Desnues C., et al., &#8220;Biodiversity and biogeography of phages in modern stromatolites and thrombolites&#8221; Nature <strong>452</strong>:340-345 (2008) (<a class="reference external" href="http://www.bio.sdsu.edu/faculty/kelley/31.pdf">http://www.bio.sdsu.edu/faculty/kelley/31.pdf</a>)</p>
<p><strong>fanning</strong> Dinsdale, EA., et al., &#8220;Microbial ecology of four coral atolls in the Northern Line Islands&#8221; PLoS One. <strong>3</strong>:e1584 (2008). (<a class="reference external" href="http://www.plosone.org/article/info:doi/10.1371/journal.pone.0001584">http://www.plosone.org/article/info:doi/10.1371/journal.pone.0001584</a>)</p>
<p><strong>wastewater</strong> Sanapareddy, N., et al., &#8220;Molecular Diversity of a North Carolina Wastewater Treatment Plant as Revealed by Pyrosequencing&#8221; Appl Environ Microbiol. <strong>75</strong>:1688–1696. (2009). (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2655459/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2655459/</a>)</p>
<p><strong>ice</strong> Simon, C., A. Wiezer, AW Strittmatter, R. Daniel, &#8220;Phylogenetic Diversity and Metabolic Potential Revealed in a Glacier Ice Metagenome&#8221; App. &amp; Env. Microb. <strong>75</strong>:7519-7526 (2009). (<a class="reference external" href="http://aem.asm.org/content/75/23/7519.full">http://aem.asm.org/content/75/23/7519.full</a>). In agreement that the major phyla are Betaproteobacteria, Bacteroidetes, and Actinobacteria.</p>
<p><strong>Lake Lanier</strong> Oh, S., et al., &#8220;Metagenomic insights into the evolution, function and complexity of the planktonic microbial community of Lake Lanier, a temperate freshwater ecosystem&#8221; App. &amp; Env. Microbiol. doi: 10.1128 (2011). (<a class="reference external" href="http://aem.asm.org/content/early/2011/07/15/AEM.00107-11.short">http://aem.asm.org/content/early/2011/07/15/AEM.00107-11.short</a>).</p>
<p><strong>Yellowstone</strong> Inskeep, WP., &#8220;Metagenomes from High-Temperature Chemotrophic Systems Reveal Geochemical Controls on Microbial Community Structure and Function&#8221; PLoS ONE <strong>5</strong>:e9773 (2010). (<a class="reference external" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0009773">http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0009773</a>).  Not sure if this is the correct study.</p>
<p><strong>biogas</strong> Jaenicke, S., et al. &#8220;Comparative and Joint Analysis of Two Metagenomic Datasets from a Biogas Fermenter Obtained by 454-Pyrosequencing&#8221; PLoS one <strong>6</strong>:e4519 (2011) (<a class="reference external" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0014519">http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0014519</a>)</p>
<p><strong>Leaf-cutter ants</strong> Aylward FO, et al. &#8220;Metagenomic and metaproteomic insights into bacterial communities in leaf-cutter ant fungus gardens&#8221; ISME J. <strong>6</strong>:1688-701 (2012). (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22378535">http://www.ncbi.nlm.nih.gov/pubmed/22378535</a>)</p>
<p><strong>kimchi</strong> Jung JY, Lee SH, Kim JM, Park MS, Bae JW, Hahn Y, Madsen EL, Jeon CO, &#8220;Metagenomic analysis of kimchi, a traditional Korean fermented food&#8221; Appl Environ Microbiol. <strong>77</strong>:2264-74 (2011). (<a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21317261">http://www.ncbi.nlm.nih.gov/pubmed/21317261</a>)</p>
<p><strong>rumen</strong> Hess, M., et al., &#8220;Metagenomic discovery of biomass-degrading genes and genomes from cow rumen&#8221; Science. <strong>331</strong>:463-7 (2011). (<a class="reference external" href="http://www.sciencemag.org/content/331/6016/463.short">http://www.sciencemag.org/content/331/6016/463.short</a>)</p>
<p><strong>Global Ocean Survey</strong> Rusch, DB, et al. &#8220;The <em>Sorcerer II</em> global ocean sampling expedition: Northwest Atlantic through Eastern Tropical Pacific&#8221; PLoS Biology <strong>5</strong>:e77(2007) (<a class="reference external" href="http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0050077">http://www.plosbiology.org/article/info:doi/10.1371/journal.pbio.0050077</a>).</p>
</div>
<div class="section" id="functional-profiles">
<h2>5.3. Functional profiles<a class="headerlink" href="#functional-profiles" title="Permalink to this headline">¶</a></h2>
<p>It is possible to produce functional profiles of the same samples used above for comparing phylogenetic profiles.  Since we are using the SEED functional classification scheme, it is also possible to &#8216;roll-up&#8217; the functional profiles into 28 categories of gene function.  Inspection of the output file SRR059330.fasta.fun reveals the following roll-up categories, defined in the notation of Fortran 90 <a class="reference download internal" href="_downloads/ruf.f90"><tt class="xref download docutils literal"><span class="pre">here</span></tt></a>.</p>
<div class="figure">
<img alt="_images/hmbm_froll.png" src="_images/hmbm_froll.png" style="width: 100%;" />
<p class="caption">Functional profile of 400 human microbiome samples, computed using the bact403 tree, the SEED classification, and Sequedex, rolled up with the above-defined classifications.  The functional profiles are provided above in an Excel spreadsheet.</p>
</div>
<p>The most striking observation about the functional profile across the 400 human microbiome samples is the consistent relative amplitude of the different functions from the different body sites.  The two sample types with the greatest variation in functional rollup profiles are the posterior fornix and the anterior nares, consistent with Figure 2 of the Nature article referenced above (Nature <strong>486</strong>:207-214 (2012)).  It is possible that this variability is a consequence of clonal populations of particular organisms that make up a significant population of some of the samples skewing the results, but we have not explored this idea further.  We observe in the environmental functional profiles, below, that these relative amplitudes are also broadly in agreement across diverse ecosystems.  We will return to this discussion below, when we examine the functional profiles in greater resolution.</p>
<div class="figure">
<img alt="_images/env4_froll.png" src="_images/env4_froll.png" style="width: 100%;" />
<p class="caption">Functional profile of 242 environmental microbiome samples with the same color-scheme as the HMB functional profile above, computed using the bact403 tree, the SEED classification, and Sequedex, rolled up with the above-defined classifications.  The functional profiles are provided above in an Excel spreadsheet.  As with the human microbiome samples, the consistent relative fraction of each functional rollup is quite striking, as is the enormous dispersions that arise from this uniformity.</p>
</div>
<div class="figure">
<img alt="_images/face_fn.png" src="_images/face_fn.png" style="width: 70%;" />
<p class="caption">Functional profile comparing soil metagenomics samples from five different FACE (<a class="reference external" href="http://public.ornl.gov/face/DUKE/duke_home.shtml">http://public.ornl.gov/face/DUKE/duke_home.shtml</a>) sites: MDE - a Maryland estuary (<a class="reference external" href="http://www.serc.si.edu/labs/co2/marsh.aspx">http://www.serc.si.edu/labs/co2/marsh.aspx</a>), NCD - a deciduous in North Carolina (<a class="reference external" href="http://public.ornl.gov/face/DUKE/duke_home.shtml">http://public.ornl.gov/face/DUKE/duke_home.shtml</a>), TNO - a plantation in Tennessee (<a class="reference external" href="http://public.ornl.gov/face/ORNL/ornl_home.shtml">http://public.ornl.gov/face/ORNL/ornl_home.shtml</a>), and a Crust (cyanobacterial soil crust) and cCreosote (root zone of creosote bush) sample in Nevada (<a class="reference external" href="http://public.ornl.gov/face/NDFF/ndff_home.shtml">http://public.ornl.gov/face/NDFF/ndff_home.shtml</a>).</p>
</div>
<p>The twenty functional categories with more than 1000 counts in the TNO sample are shown below with the subsystem number from the figure and the three levels of annotation.  The top two levels are shortened to help make the lowest level fit on the page.  Not shown, but available in the appropriate file, <tt class="xref download docutils literal"><span class="pre">subsystems2roles</span></tt> is the list of particular genes included in each subsystem.  Many of the subsytems have detailed explanations available at <a class="reference external" href="http://www.theseed.org">http://www.theseed.org</a>.</p>
<div class="highlight-python"><pre>si_0032 Amino Acids    Glutamine, GLN, ASP, ASN; ammonia    Glutamine,_Glutamate,_Aspartate_and_Asparagine_Biosynthesis
si_0042 Amino Acids    Lysine, threonine, MET, Cys          Methionine_Biosynthesis
si_0078 Carbohydrates  Central carbohydrate metabolism      Pyruvate_metabolism_II:_acetyl-CoA,_acetogenesis_from_pyruvate
si_0122 Carbohydrates  One-carbon Metabolism                Serine-glyoxylate_cycle
si_0205 Cell Wall      Unclassified cell wall and capsule   Peptidoglycan_Biosynthesis
si_0279 Clustering     Unclassified clustering-based        Bacterial_Cell_Division
si_0334 Clustering     Unclassified clustering-based        Conserved_gene_cluster_associated_with_Met-tRNA_formyltransferase
si_0357 Cofactors      Biotin                               Biotin_biosynthesis
si_0415 DNA Meta.      DNA replication                      DNA-replication
si_0448 Fatty Acids    Fatty acids                          Fatty_Acid_Biosynthesis_FASII
si_0526 Membrane tra.  Unclassified membrane transport      Ton_and_Tol_transport_systems
si_0589 Nitrogen       Unclassified nitrogen metabolism     Ammonia_assimilation
si_0602 Nucleosides    Purines                              Purine_conversions
si_0640 Phosphorus     Unclassified phosphorus metabolism   Phosphate_metabolism
si_0654 Potassium      Unclassified potassium metabolism    Potassium_homeostasis
si_0660 Protein        Protein biosynthesis                 Ribosome_LSU_bacterial
si_0790 Regulation     Unclassified regulation and cell sig cAMP_signaling_in_bacteria
si_0812 Respiration    Electron donating reactions          Respiratory_Complex_I
si_0929 Virulence      Resistance to antibiotics and toxic  Cobalt-zinc-cadmium_resistance
si_0939 Virulence      Resistance to antibiotics and toxic  Multidrug_Resistance_Efflux_Pumps</pre>
</div>
<p>Perhaps the most striking aspect of this plot is the sample-to-sample consistency in the fraction of reads recruited to each functional category.  To investigate further, in the next section, we examine the functional profiles of the larger data sets for the 400 human microbiome samples and 242 enviromental samples for individual SEED subsystems, answering the question, &#8216;What type of nitrogen metabolism is present?&#8217; rather than &#8216;Does the organism metabolise nitrogen?&#8217;.</p>
</div>
<div class="section" id="normalized-functional-profiles">
<h2>5.4. Normalized functional profiles<a class="headerlink" href="#normalized-functional-profiles" title="Permalink to this headline">¶</a></h2>
<p>Since most of the SEED subsystems cover metabolic and other functions necessary for all organisms, the consistency across samples is more reassuring than informative.  To convince ourselves that the functional profiles are capable of distinguishing samples from one-another, we normalize the functional counts (not including &#8216;unassigned function&#8217; and compare two categories that are observed to vary across environmental and human microbiome samples.</p>
<p>Normalization occured with Fortran 90 code; see <em class="xref std std-ref">F90 code</em>.  One of the easiest functional categories to interpret is photosynthesis across the environmental metagenomes.  By comparison, none of the human microbiome samples contained more than 0.25% photosynthesis, and this was primarily composed of hits to proteorhodopsin, rather than the photosystems.</p>
<div class="figure">
<img alt="_images/env4_photosynthesis.png" src="_images/env4_photosynthesis.png" style="width: 100%;" />
<p class="caption">Fraction of the functionally assigned reads encoding for phytosynthesis subsystems across the 242 environmental metagenomes.</p>
</div>
<div class="figure">
<img alt="_images/hmbm_motility.png" src="_images/hmbm_motility.png" style="width: 100%;" />
<p class="caption">Fraction of the functionally assigned reads encoding for phytosynthesis subsystems across the 400 human microbiome metagenomes.</p>
</div>
<p>The set of output files can be viewed in Excel, and we supply two Excel spreadsheets with the collected results for the human microbiome data <tt class="xref download docutils literal"><span class="pre">hmb.xlsx</span></tt> and environmental metagenomes <tt class="xref download docutils literal"><span class="pre">env4.xlsx.</span></tt>   The tab &#8216;fnorm&#8217; in each spreadsheet contains the normalized functional counts.  It is straightforward to scroll through this spreadsheet graphing various subsystems of interest, or sorting the functional categories based on counts in particular samples or difference in counts across various samples.  With some thought, it is also possible to sort based on the signal-to-noise ratio with which the functional profiles distinguish ecosystems, with the noise defined by replicate samples.</p>
<p>Thus, normalized functional profiles across the two reference data sets enable a &#8216;top-down&#8217; approach to understanding functional classifications.  The user may want to import results from their own metagenomics data sets into these spreadsheets to better understand the significance of their own results.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">5. Phylogenetic and functional profiles</a><ul>
<li><a class="reference internal" href="#acquiring-data-files">5.1. Acquiring data files</a></li>
<li><a class="reference internal" href="#phylogenetic-rollups">5.2. Phylogenetic rollups</a></li>
<li><a class="reference internal" href="#functional-profiles">5.3. Functional profiles</a></li>
<li><a class="reference internal" href="#normalized-functional-profiles">5.4. Normalized functional profiles</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="starting.html"
                        title="previous chapter">4. Getting started</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="tree_of_life.html"
                        title="next chapter">6. Tree of Life</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/profiles.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="tree_of_life.html" title="6. Tree of Life"
             >next</a> |</li>
        <li class="right" >
          <a href="starting.html" title="4. Getting started"
             >previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; <a href="copyright.html">Copyright</a> 2012, Los Alamos National Security, LLC.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>