

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>4. Getting started &mdash; Sequedex 0.9.2 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.9.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/tree.ico"/>
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="Sequedex 0.9.2 documentation" href="index.html" />
    <link rel="next" title="5. Phylogenetic and functional profiles" href="profiles.html" />
    <link rel="prev" title="3. Installation instructions" href="installation.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="profiles.html" title="5. Phylogenetic and functional profiles"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="installation.html" title="3. Installation instructions"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="getting-started">
<h1>4. Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<p>Sequedex can be run in several modes with either a simple graphical user interface or directly from the command line.  The platform-specific instructions for starting the graphical user interface and obtaining a license file were provided in the previous chapter, and a description of command line use of Sequedex is provided at the end of this chapter.  If the license file is not obtained or installed incorrectly, sequedex will process a limited number of reads (~10,000) and work only in single-thread mode.  For the bulk of this chapter, we will assume the graphical user interface is being used, and that the user has some way of looking at the tab-delimited text output files, whether that be Excel, a text editor, or an analysis environment like R or Matlab.</p>
<p>Sequedex is distributed with several sets of test data for demonstration and validation.  In this chapter, we illustrate how to use Sequedex on these datasets.   Two metagenomics data files are provided, one from a stool sample sequenced under the human microbiome program and two from a soil sample collected from a Maryland estuary.  Also provided are synthetic data files constructed from the draft genomes of four novel organisms cultured from desert cyanobacterial mats, <em>Bacillus mojavensis, Bosea thiooxidans, Herbaspirillum seropedicae,</em> and <em>Microbacterium trichotecenolyticum.</em>  The complete stool sample file can be obtained at <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser&amp;run=SRR059346">http://www.ncbi.nlm.nih.gov/Traces/sra/?view=run_browser&amp;run=SRR059346</a> and the Maryland estuary sample analysis can be compared to the analysis in Figure 7 of Berendzen, et al. <a class="footnote-reference" href="#s1" id="id1">[1]</a> as well as additional files 5 and 6 of this work.  The synthetic data sets were used in this same manuscript to explore the read-length dependence of Sequedex&#8217;s sensitivity and specificity, illustrated in Figure 6.</p>
<p>The test data set for the human microbiome sample is located in the directory sequedex/testData/metagenomes/humanMicrobiome/gut/.  If you are starting the Sequedex GUI from the command line, it may be easiest to navigate this directory before typing sequescan.  Alternatively, you can start the sequescan GUI from the applications menu and navigate to the data directory with the dialog box.  In either case, you should see a screen that looks like this when you start the Sequedex application:</p>
<div class="figure">
<img alt="_images/sequescan_gui.png" src="_images/sequescan_gui.png" style="width: 60%;" />
<p class="caption"><strong>The graphical user interface for Sequescan.</strong>  At the top are two pull-down menus that enable the user to request and install a license and to quit the program.  Three input selections allow the user to choose the signature data module, the functional data module and select whether the fasta input files will be chosen by name, by directory, or by an input file listing the complete path of files to be analyzed.  The &#8216;Input&#8217; box is used to select a data file, data directory, or a text file listing data files to be analyzed, depending on &#8216;Input type&#8217; selected above.  A configuration file is selected to specify additional program options.  The top-level directory to which output files will be written can be selected.  Finally, the number of threads used and whether verbose output messages are desired can be selected.  Note that at present multiple threads can only be used when multiple files are simultaneously analyzed; each file is assigned a thread.</p>
</div>
<p>Click on the button to the right of the &#8216;input&#8217; field and locate the human microbiome file &#8216;SRR059346-first100kReads.fasta&#8217;.  After opening this file and clicking the &#8216;Run Sequescan&#8217; button at the bottom of the dialog box, Sequescan will begin reading the bacterial signature list into memory, then classifying the reads in SRR059346-first100kReads.fasta against this signature list.  Output similar to that below will appear in the &#8216;Progress&#8217; box at the bottom of the GUI.:</p>
<div class="highlight-python"><pre>Sep 14, 2012 11:58:01 AM  Start Sequescan Program (mode=run)
Sep 14, 2012 11:58:01 AM  Reading signature map from data module
Sep 14, 2012 11:59:13 AM  Reading functions from data module
Sep 14, 2012 11:59:13 AM  Start processing ~mcmahon/sequedex/testData/metagenomes/humanMicrobiome/gut/SRR059346-first100kReads.fasta
Sep 14, 2012 11:59:28 AM  SRR059346-first100kReads.fasta: 26348176 chars processed (100.0 completed)
Sep 14, 2012 11:59:33 AM  End Sequescan Program</pre>
</div>
<p>The output files will appear in subdirectories &#8216;log&#8217; and &#8216;data&#8217; in the directory where the input file was (if you used the default settings).  The log directory will contain a subdirectory with an html file that can be opened with a web-browser, indicating any errors or warnings associated with this Sequescan run.  The data directory will contain a subdirectory with four types of outputs in two seperate formats (tab-separated-variables and json / tsvj); we examine the four types of tsv files in the next section.  These files can be imported into Excel, a text editor, or simply examined on the command line with the &#8216;less&#8217; or &#8216;cat&#8217; commands.</p>
<div class="section" id="output-files-stats">
<h2>4.1. Output files - stats<a class="headerlink" href="#output-files-stats" title="Permalink to this headline">¶</a></h2>
<p>The output file SRR059346-first100kReads.fasta-bact403.1-stats.tsv contains basic statistics about the computation. The stats file resulting for the above run contains:</p>
<div class="highlight-python"><pre>Data Input File /home/localhost/mcmahon/sequedex/testData/metagenomes/humanMicrobiome/gut/SRR059346-first100kReads.fasta
Percent of File Processed                    100.0
Processing Complete                          true
Thread Pool Size                             1
Processing Rate (Gbp/hr)                     4.761
Reads In                                     100000
Bases In                                     20200000
Reads With Fragments &gt; 15 Bases              99347
Frames Processed                             600000
Frames with Fragments &gt; 15 Bases             591859
Fragments In                                 2388546
Fragments &gt; 15 Bases                         931875
Reads Assigned                               69170
Single-Signature Reads                       2117
Single-Node Reads                            16035
Monophyletic Reads                           49835
Non-Monophyletic Reads                       1183
Fragments Assigned                           123589
Percent of Reads Assigned                    69.2
Total Size of Matched Fragments (bp)         18635964
Fragments Assigned                           93977
Total Size of Fragments Assigned (bp)        14288283
Input Time(ms)                               170
Translate Time (ms)                          9060
Match Time (ms)                              4993
Assignment Time (ms)                         297
Other Time (ms)                              753
Total Time (ms)                              15273</pre>
</div>
<p>The input file is listed on the top line, followed by the fraction of the file processed, which in this case is 100%.  For large files, intermediate results are written to output files, and this line indicates an estimated fraction processed.  In the case of a missing or invalid license file, only a fraction is prcessed, and that fraction would be indicated here.  The &#8216;Processing Complete&#8217; line indicates &#8216;true&#8217; if the processing completed normally, whether because the end of the input file was reached or because a valid license was not found.</p>
<p>The Thread Pool Size of one is indicated, together with a processing rate of 4.761 Gbp/hr.  When processing multiple files simultaneously, it is possible to have a dedicated Java thread processing each file, all comparing to the same memory map.  The processing rate provided is of the single thread dedicated to this particular file.  Typical values for computers in use as of this writing are 3.5 to 7 Gbp/hr, with the rate per thread dropping roughly a factor of two when 40-60 processors are running together.  Thus a 64 core machine capable of 3.7 Gbp/hr on a single thread was able to process fasta files at the cumulative rate of 100 Gbp/hr, and each output file had a processing rate of 1.7 Gbp/hr.</p>
<p>The 100,000 reads and 20,200,000 base pairs processed are indicated next, from which the average read length of 202 base pairs can be computed.  Of the 100,000 reads processed, 99,347 contained at least one region with 15 or more consecutive amino acids uninterrupted by an undetermined nucleotide or stop codon.  This cutoff value of 15 amino acides necessary for further consideration by Sequedex can be changed in the configuration file.  Sequedex at present does not recognize ambiguity codons.</p>
<p>From the 100,000 reads, 600,000 reading frames were processed, of which 591,859 contained an amino acid region with 15 or more concecutive amino acids.  Since many of the reads contained mulitple fragments, 2,388,546 peptide fragments were identified, of which 931,895 (less than half) were of length 15 amino acids or longer.</p>
<p>Over-all, 69,170 reads (69% of the processed reads) contained at least one signature peptide and thus could be assigned a phylogenetic placement.  This number is in line with expections from the synthetic data described in Berendzen et al. <em>BMC Res. Rep.</em> <strong>5</strong> :460 (2012) for 200 base pair reads from a bacteria similar to those in the reference data base.  The 69,170 reads are sub-divided into four types: 2,117 reads contained only one signature; 16,035 contained more than one signature, but all signatures were from the same node; 49,835 reads, or 72% of those assigned were monophyletic, or contained signatures assigned along a single path from the root of the reference phylogeny to a particular set of leaves; 1,183 assigned reads were non-monophyletic, or contained at least two signatures that were not on a single path from the root to a particular set of leaves on the reference tree.</p>
<p>A total of 123,589 peptide fragments were assigned a phylogeny; these fragments came from the 69.2% of the reads with an assignable fragment, and consisted of 18,635,964 base pairs. A total of 93,977 peptide fragments were also assigned a function; these fragments consisted of 14,288,283 base pairs.</p>
<p>The next six lines break the total processing time (15.2 seconds) into input, translating, matching, assignment steps, as well as &#8216;other&#8217;.</p>
</div>
<div class="section" id="output-files-phylogeny-counts">
<h2>4.2. Output files - phylogeny counts<a class="headerlink" href="#output-files-phylogeny-counts" title="Permalink to this headline">¶</a></h2>
<p>The output file SRR059346-first100kReads.fasta-bact403.1-phylogeny_counts.tsv contains the phylogenetic profile derived from the fasta input file.  It contains eight columns of data, containing node-numbers defined with respect to a reference phylogeny, labels for these nodes, and histograms of reads assigned to these nodes in several different ways. Several sections of this output file are shown below, with the majority of reads recruiting to node 189 in the bacterial reference tree, consistant with a clonal population of <em>E. coli</em> in this patient::</p>
<div class="highlight-python"><pre>indx    id      rollup_category             total  single_sig  single_node  monophyl.  non_monophyl
0       n000    Root                         1685     187          764          0         734
1       n001    Multi-Phyla                    55       9           28         18          0
2       n002    Cyanobacteria+Firmicutes       23       5           12          6          0
3       n003    Cyanobacteria+Firmicutes        1       0            0          1          0
4       n004    Cyanobacteria                   1       1            0          0          0
...
25      n025    Firmicutes                      3       1            0          2          0
26      n026    Firmicutes                     24      15            1          8          0
27      n027    Firmicutes                     12       1            3          8          0
28      n028    Firmicutes                      2       0            0          2          0
29      n029    Firmicutes                     22       1            6         15          0
30      n030    Firmicutes                     38       1           16         21          0
31      n031    Firmicutes                     12       1            2          9          0
32      n032    Firmicutes                     12       1            3          8          0
33      n033    Firmicutes                      2       0            1          1          0
...
184     n184    Beta+Gammaproteobacteria       17       1            0         16          0
185     n185    Beta+Gammaproteobacteria       16       0            0         16          0
186     n186    Beta+Gammaproteobacteria       14       1            1         12          0
187     n187    Beta+Gammaproteobacteria       42       8            2         32          0
188     n188    Beta+Gammaproteobacteria     1567      60          108       1399          0
189     n189    Beta+Gammaproteobacteria    46469     628         9071      36770          0
190     n190    Beta+Gammaproteobacteria      432      12           71        349          0
191     n191    Beta+Gammaproteobacteria       14       1            2         11          0
192     n192    Beta+Gammaproteobacteria       28       0            0         28          0
193     n193    Beta+Gammaproteobacteria        2       0            0          2          0</pre>
</div>
<p>The columns of the phylogenetic profile are:</p>
<ul class="simple">
<li>indx:   Index starting at 0</li>
<li>id:     node id. consistent with  <a class="reference download internal" href="_downloads/Bacteria403.newick"><tt class="xref download docutils literal"><span class="pre">newick</span></tt></a>, and <a class="reference download internal" href="_downloads/Bacteria4031.phyloxml"><tt class="xref download docutils literal"><span class="pre">phyloxml</span></tt></a> reference tree.</li>
<li>rollup_category: general category name.  May include several phylum for nodes near the root</li>
<li>total:   Total number of reads attributed to the node</li>
<li>single_sig:  Number of reads containing only one signatures</li>
<li>single_node:  Number of reads containing multiple signatures, but all of them belonging to the same node</li>
<li>monophyletic: Number of reads containing multiple signatures that are consistent along the phylogeny</li>
<li>non_monophyl: Number of reads containing multiple signatures that are inconsistent with the phylogeny</li>
</ul>
<p>Visualization of the phylogenetic profiles can be aided by placing the counts onto the phylogenetic
tree <a class="reference download internal" href="_downloads/Bacteria403.newick"><tt class="xref download docutils literal"><span class="pre">newick</span></tt></a>, or <a class="reference download internal" href="_downloads/Bacteria4031.phyloxml"><tt class="xref download docutils literal"><span class="pre">phyloxml.</span></tt></a>  See
Chapter 6 for how to do this.</p>
</div>
<div class="section" id="output-files-function-counts">
<h2>4.3. Output files - function counts<a class="headerlink" href="#output-files-function-counts" title="Permalink to this headline">¶</a></h2>
<p>Functional profiles are computed using a reference database of functional categories.  The default functional classification is derived from that used by the SEED project (<a class="reference external" href="http://www.theseed.org">http://www.theseed.org</a>) database as described in detail in Ref. <a class="footnote-reference" href="#s1" id="id2">[1]</a>  The functional profiles count the number of open reading frames associated to each functional category.  Because many signatures occur in multiple SEED categories and each reading frame gets a total of one count, categories often are assigned a fractional count.</p>
<p>Approximately 70% of the phylogenetic signatures can be associated with one or more SEED subsystems.</p>
<p>The output file SRR059346-first100kReads.fasta-bact403.1-seed_0911-function_counts.tsv contains the functional profile derived from the fasta input file. The carbohydrate section of
this file is shown here::</p>
<div class="highlight-python"><pre>indx    fragments  id      level1          level2              subsystem
54      0.5     si_0054 Carbohydrates   Aminosugars     (GlcNAc)2_Catabolic_Operon
55      285     si_0055 Carbohydrates   Aminosugars     Chitin_and_N-acetylglucosamine_utilization
56      195.3   si_0056 Carbohydrates   Aminosugars     Fructoselysine_(Amadori_product)_utilization_pathway
57      355.7   si_0057 Carbohydrates   Aminosugars     N-Acetyl-Galactosamine_and_Galactosamine_Utilization
58      1       si_0058 Carbohydrates   Aminosugars     Neotrehalosadiamine_(NTD)_Biosynthesis_Operon
59      6.5     si_0059 Carbohydrates   CO2 fixation    CO2_uptake,_carboxysome
60      217.4   si_0060 Carbohydrates   CO2 fixation    Calvin-Benson_cycle
61      6       si_0061 Carbohydrates   CO2 fixation    Carboxysome
62      184.6   si_0062 Carbohydrates   CO2 fixation    Photorespiration_(oxidative_C2_cycle)
63      139.5   si_0063 Carbohydrates   Central carbohydrate metabolism Dehydrogenase_complexes
64      134.5   si_0064 Carbohydrates   Central carbohydrate metabolism Dihydroxyacetone_kinases
65      315.9   si_0065 Carbohydrates   Central carbohydrate metabolism Entner-Doudoroff_Pathway
66      2       si_0066 Carbohydrates   Central carbohydrate metabolism Ethylmalonyl-CoA_pathway_of_C2_assimilation
67      158.3   si_0067 Carbohydrates   Central carbohydrate metabolism Glycolate,_glyoxylate_interconversions
68      385.5   si_0068 Carbohydrates   Central carbohydrate metabolism Glycolysis_and_Gluconeogenesis
69      41      si_0069 Carbohydrates   Central carbohydrate metabolism Glycolysis_and_Gluconeogenesis,_including_Archaeal_enzymes
70      240.8   si_0070 Carbohydrates   Central carbohydrate metabolism Glyoxylate_bypass
71      71.4    si_0071 Carbohydrates   Central carbohydrate metabolism Methylglyoxal_Metabolism
72      0       si_0072 Carbohydrates   Central carbohydrate metabolism Particulate_methane_monooxygenase_(pMMO)
73      212.2   si_0073 Carbohydrates   Central carbohydrate metabolism Pentose_phosphate_pathway
74      2.9     si_0074 Carbohydrates   Central carbohydrate metabolism Peripheral_Glucose_Catabolism_Pathways
75      44.3    si_0075 Carbohydrates   Central carbohydrate metabolism Pyruvate:ferredoxin_oxidoreductase
76      279     si_0076 Carbohydrates   Central carbohydrate metabolism Pyruvate_Alanine_Serine_Interconversions
77      344.9   si_0077 Carbohydrates   Central carbohydrate metabolism Pyruvate_metabolism_I:_anaplerotic_reactions,_PEP
78      310.9   si_0078 Carbohydrates   Central carbohydrate metabolism Pyruvate_metabolism_II:_acetyl-CoA,_acetogenesis_from_pyruvate
79      0       si_0079 Carbohydrates   Central carbohydrate metabolism Soluble_methane_monooxygenase_(sMMO)
80      387.1   si_0080 Carbohydrates   Central carbohydrate metabolism TCA_Cycle
81      345.6   si_0081 Carbohydrates   Di- and oligosaccharides        Beta-Glucoside_Metabolism
82      172.5   si_0082 Carbohydrates   Di- and oligosaccharides        Fructooligosaccharides(FOS)_and_Raffinose_Utilization
83      317     si_0083 Carbohydrates   Di- and oligosaccharides        Lactose_and_Galactose_Uptake_and_Utilization
84      202.4   si_0084 Carbohydrates   Di- and oligosaccharides        Lactose_utilization
85      1023.2  si_0085 Carbohydrates   Di- and oligosaccharides        Maltose_and_Maltodextrin_Utilization
86      147.6   si_0086 Carbohydrates   Di- and oligosaccharides        Melibiose_Utilization
87      48.3    si_0087 Carbohydrates   Di- and oligosaccharides        Sucrose_utilization
88      1       si_0088 Carbohydrates   Di- and oligosaccharides        Sucrose_utilization_Shewanella
89      210.5   si_0089 Carbohydrates   Di- and oligosaccharides        Trehalose_Biosynthesis
90      407.9   si_0090 Carbohydrates   Di- and oligosaccharides        Trehalose_Uptake_and_Utilization
91      4.8     si_0091 Carbohydrates   Di- and oligosaccharides        Unknown_oligosaccharide_utilization_Sde_1396
92      72.9    si_0092 Carbohydrates   Fermentation    Acetoin,_butanediol_metabolism
93      154.2   si_0093 Carbohydrates   Fermentation    Acetone_Butanol_Ethanol_Synthesis
94      284.4   si_0094 Carbohydrates   Fermentation    Acetyl-CoA_fermentation_to_Butyrate
95      212     si_0095 Carbohydrates   Fermentation    Butanol_Biosynthesis
96      54.2    si_0096 Carbohydrates   Fermentation    Fermentations:_Lactate
97      420.9   si_0097 Carbohydrates   Fermentation    Fermentations:_Mixed_acid
98      71.5    si_0098 Carbohydrates   Glycoside hydrolases    Predicted_carbohydrate_hydrolases</pre>
</div>
<p>The columns of this file are::</p>
<div class="highlight-python"><pre>*   indx :  Index, starting at zero
*   fragments: Number of fragment assigned to that SEED category
*   id : SEED identification number
*   level1 : Level 1 SEED category descriptor
*   level2 : Level 2 SEED category descriptor
*   subsystem : Level 3 or subsystem SEED category descriptor</pre>
</div>
<p>More information about particular SEED subsystems can be obtained at <a class="reference external" href="http://www.theseed.org/SubsystemStories">http://www.theseed.org/SubsystemStories</a>, but the user should note that the index numbers above (eg. si_0094) are only defined within the Sequedex distribution.</p>
</div>
<div class="section" id="output-files-who-does-what-matrix">
<h2>4.4. Output files - who does what matrix<a class="headerlink" href="#output-files-who-does-what-matrix" title="Permalink to this headline">¶</a></h2>
<p>The output file SRR059346-first100kReads.fasta-bact403.1xseed_0911.tsv tabulates the number of reads in each phylogeny <strong>and</strong> functional category.  It is a matrix with 402 columns and 911 rows, corresponding to the phylogeny vector defined by the reference phylogeny and the functional vector defined by the SEED subsystems.  A portion of this matrix is shown below, centered around the enteric bacteria and the carbohydrate metabolism subsystems::</p>
<div class="highlight-python"><pre>   id   ...  n184 n185 n186 n187 n188   n189  n190  n191 n192 n193  ...
...
si_0054        0    0    0    0    0     0.5    0    0    0    0
si_0055        0    0    0    0    6.6  200.6  0.1  0.2   0    0
si_0056        0    0    0    0    47   147.3   0    0    0    0
si_0057        0    0    0    0    0    283.4   0    0    0    0
si_0058        0    0    0    0    0      0     0    0    0    0
si_0059        0    0    0    0    0     2.9    0    0    0    0
si_0060        0   0.1   0    0   1.1   117.1  0.5   0    0    0
si_0061        0    0    0    0    0     3.8    0    0    0    0
si_0062        0    0    0    0    4    99.8    0    0    0    0
si_0063        0    0    0    0   6.8   72.6    0    0    0    0
si_0064        0    0    0    0   0.3   124.2   0    0    0    0
si_0065        0   0.1   0    0   7.9   199.8   0    0    0    0
si_0066        0    0    0    0    0     0.7    0    0    0    0
si_0067        0    0    0    0   0.7   95.2    0    0    0    0
si_0068        0   0.1   0    0   9.1   227.1   0    0    0    0
si_0069        0    0    0    0   0.9   15.6    0    0    0    0
si_0070        0    0    0    0   9.3   143.1   0    0    0    0
si_0071        0    0    0    0    1     51.5  0.3   0    0    0
si_0072        0    0    0    0    0      0     0    0    0    0
si_0073        0    0    0    0   8.6   106.6  0.5   0    0    0
si_0074        0    0    0    0    0     1.6    0    0    0    0
si_0075        0    0    0    0    1     26     0    0    0    0
si_0076        0    0    1    0   8.4   190     0    0    0    0
si_0077        0    0    2    0   3.9   174.5   6    0    0    0
si_0078        0    0    0    0  15.8   191.7  0.3   0    0    0
si_0079        0    0    0    0   0       0     0    0    0    0
si_0080        0    0    0    0   9.1   223.4   1    0    0    0
si_0081        0    0   0.1   0   8.6   228.2   0    0    0    0
si_0082        0    0    0    0   7.2   120.3  2.2   0    0    0
si_0083        0    7   0.1   6  11.1   168.9 15.3   0   0.3   0
si_0084        0    0    0    0   2.2   175.1   0    0    0    0
si_0085        0    0    0    0  87.2   749    2.2  0.2   0    0
si_0086        0    0    0    0   3.5   142.6   0    0    0    0
si_0087        0    0    0    0    0     38.8  6.9   0    0    0
si_0088        0    0    0    0    0      0.5   0    0    0    0
si_0089        0    0    0    0  11.3   158.8   0    0    0    0
si_0090        0    0    0    0  12.4   310.2   0    0    0    0
si_0091        0    0    0    0    0      3.3   0    0    0    0
si_0092        0    0    0    0  0.5     29.2  2.5   0    0    0
si_0093        0    0    0    0    0    112.4   0    0    0    0
si_0094        0    0    0    0   1.6   224.2   0    0    0    0
si_0095        0    0    0    0   3.4   153.1   0    0    0    0
si_0096        0    0    0    0   2.2    30.2   0    0    0    0
si_0097        0    0    0    0  13.8   302.1   0    0    0    0
si_0098        0    0    0    0    0     70.5   0    0    0    0
...</pre>
</div>
<p>Generally, this will be most useful on extremely large data sets, as the counts are spread across approximately 400,000 bins.  Given the even increasing number of reads generated by modern sequencers, we expect to have available sufficiently many reads to provide statistical resolving power.  As discussed in the phylogeny and the functional profile output files, the results must be put into context with the used databases.</p>
</div>
<div class="section" id="analysis-of-multiple-files-in-parallel-with-the-gui">
<h2>4.5. Analysis of multiple files in parallel with the GUI<a class="headerlink" href="#analysis-of-multiple-files-in-parallel-with-the-gui" title="Permalink to this headline">¶</a></h2>
<p>We demonstrate the ability of Sequedex to process multiple fasta files in parallel, as a single batch.  Besides eliminating the need to individually specify each fasta file with the GUI, batch processing of fasta files results in a significant speedup of computation time, because the signature list being read only once, and enables parallel processing of the fasta files with individual Java threads.  This not only enables you to process files simultaneously if your computer has multple cores, but it also requires the signature file be read in only onces for all of the files.</p>
<p>A set of sixteen synthetic metagenomic datasets were used in Berendzen et al. <a class="footnote-reference" href="#s1" id="id3">[1]</a> to examine the read-length of the sensitivity and specificity of Sequedex on four separate clonal populations of bacteria.  These files are provided with the Sequedex distribution in the testData directory under the &#8216;synthetic&#8217; subdirectory.  To process these files together, we can select the &#8216;directory&#8217; option from the &#8216;Input type&#8217; pull-down menu, locate the directory these fasta files reside in the &#8216;Input&#8217; dialog box, and enter &#8216;4&#8217; or more in the box by the &#8216;Thread Number&#8217; option.  Upon clicking on &#8216;Run Sequedex&#8217;, the files will be processed in a batch, with the results appearing in separate subdirectories under &#8216;data/output&#8217;.</p>
<p>Alternatively, the files to be analyzed can be listed, with their complete paths, in an input file which is selected in conjunction with the &#8216;list file&#8217; option under &#8216;input type&#8217; dialog box.  On a Linux or Mac platform, it is possible to generate a complete path by including the full path in the &#8216;ls&#8217; command as follows (which must conform to the path relevant to your own system):</p>
<div class="highlight-python"><pre>ls -1 /home/username/sequedex/testData/synthetic/*.fna</pre>
</div>
</div>
<div class="section" id="analysis-of-e-coli-proteome">
<h2>4.6. Analysis of <em>E. coli</em> proteome<a class="headerlink" href="#analysis-of-e-coli-proteome" title="Permalink to this headline">¶</a></h2>
<p>One way to validate Sequedex is to provide an input file derived from a single organism and observed the phylogenetic profile produced by Sequedex.  Since Sequedex requires the input file to be nucleotide sequence, one cannot at present input amino acid sequences.  Six-frame translations are automatically performed on input files, which must be sequeces of A, T, G, and C.  NCBI does, however, distribute files with nucleotide sequences of each of the genes in their completed and draft genomes at their ftp site.  We provide here instructions for analyzing <em>E. coli,</em> but they are readily extensible to any of the other genomes available from the same location.</p>
<p>To see the phylogenetic profile Sequedex produces from the <em>E. coli</em> genome, download the ffn file from NCBI, at <a class="reference external" href="ftp://ftp.ncbi.nih.gov/genomes/Bacteria/Escherichia_coli_K_12_substr__DH10B_uid58979/NC_010473.ffn">ftp://ftp.ncbi.nih.gov/genomes/Bacteria/Escherichia_coli_K_12_substr__DH10B_uid58979/NC_010473.ffn</a> and run Sequedex on it, as described above.  You should observe the phylogenetic profile is heavily weighted towards node 189, which is the most specific node covering <em>E. coli</em> in the reference phylogeny.  Once you are familiar with this process, it is easy to substitute any of the 1400 other completed bacterial genomes available at this ftp site.</p>
<p>Alternatively, the user can download a tar file with the files of nucleotide sequences coding for genes in each of the 1400 completed genomes at <a class="reference external" href="ftp://ftp.ncbi.nih.gov/genomes/Bacteria/all.ffn.tar.gz">ftp://ftp.ncbi.nih.gov/genomes/Bacteria/all.ffn.tar.gz</a>.  Running various individual proteomes through Sequedex will give the user some indication of the precision and sensitivity with which genes can be classified.  Note that this is an excellent situation to use the &#8216;list file&#8217; under &#8216;input type&#8217; as the ffn files will be located in directories corresponding to the organism.  Sequedex will separate the output files in a similar directory structure under the directory specified under &#8216;Output (Top-Level)&#8217;</p>
</div>
<div class="section" id="running-sequedex-from-the-command-line">
<h2>4.7. Running Sequedex from the command line<a class="headerlink" href="#running-sequedex-from-the-command-line" title="Permalink to this headline">¶</a></h2>
<p>Many users will find it advantageous to run Sequedex from the command line, which is availble to all users of Mac and Linux systems, and can be obtained for Windows by installing the Cygwin package <a class="reference external" href="http://www.cygwin.com">http://www.cygwin.com</a>.</p>
<p>However you acquire your command line, it is most convenient to change directories to where your data resides, the execute the particular Sequedex run.  Help with the basic syntax can be obtained by typing:</p>
<div class="highlight-python"><pre>sequescan run -h</pre>
</div>
<p>which provides the following output:</p>
<div class="highlight-python"><pre>sequescan run [-h] | [-v] [-c config_file] -d SIGDBNAME [-o output_directory] [-s function_set] [-t thread_num] [-l] INFILE

Example:
sequescan run -v -c myanalysis.conf  -d Bact403 -o myOutpptDir -s seed_0911 inFile

Parameter definitions:
-h    help
-v    verbose
-c    user-defined configuration file (overrides system configuation file)
-d    name of data module
-o    user-defined top-level directory for data output (overrides system-defined location)
-s    name of function set
-t    maximum number of threads in threadpool (default = 1)
-l    indicates that INFILE contains a list of fasta files to be processed</pre>
</div>
<p>Thus, running Sequescan from the directory with the fasta files with the command line:</p>
<div class="highlight-python"><pre>sequescan run -v -d bact403.1 -s seed_0911 -o out_dir infile.fas</pre>
</div>
<p>will run Sequescan with the bact403 signature list and seed_0911 functional classification of the signatures distributed with this release.  Sequescan will match all of the reads in <em>infile.fas</em> and create the directory out_dir in which it will place the results (in a subdirectory).</p>
<p>Alternatively, the command line:</p>
<div class="highlight-python"><pre>sequescan run -v -t 4 -d bact403.1 -s seed_0911 -o out_dir .</pre>
</div>
<p>will run Sequescan as before, except processing all of the files with an appropriate file extension (by default, fasta, fst, fna, fas, and ffn, specified in sequedex/etc/sequescan/sequescan.conf) as a group, using 4 Java threads, and again with the output appearing in subdirectories of out_dir.  The output files can be written to a different directory tree if one or both directories are supplied with appropriate paths.  This enables users to leave shared data directories untouched by output files produced by Sequedex without the need for recopying potentially large input files.</p>
<p>This remaining of this section describes the generated results from sequescan, which includes a phylogenetic profile, a functional profile and a matrix of function x phylogeny that informs on &#8220;who does what&#8221;.  Visualization and interpretation of these profiles are discussed in the Chapter 5 &#8220;Phylogenetic and Functional Profiles&#8221; and statistical analysis comparing multiple profiles are described in Chapter 6 &#8220;Analysis Examples&#8221;.</p>
</div>
<div class="section" id="sequedex-configuration-file">
<h2>4.8. Sequedex configuration file<a class="headerlink" href="#sequedex-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>A configuration file, sequedex/etc/sequescan/sequescan.conf, contains a variety of settings controlling default behavior of inputs and outputs of Sequescan.</p>
<table class="docutils footnote" frame="void" id="s1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[1]</td><td><em>(<a class="fn-backref" href="#id1">1</a>, <a class="fn-backref" href="#id2">2</a>, <a class="fn-backref" href="#id3">3</a>)</em> Berendzen, J., WJ Bruno, JD Cohn, NW Hengartner, CR Kuske, BH McMahon, MA Wolinsky, G Xie, &#8220;Rapid phylogenetic and functional classification of short genomic fragments with signature Peptides&#8221;, BMC Research Notes, August, 2012 (<a class="reference external" href="http://www.biomedcentral.com/1756-0500/5/460/">http://www.biomedcentral.com/1756-0500/5/460/</a>)</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">4. Getting started</a><ul>
<li><a class="reference internal" href="#output-files-stats">4.1. Output files - stats</a></li>
<li><a class="reference internal" href="#output-files-phylogeny-counts">4.2. Output files - phylogeny counts</a></li>
<li><a class="reference internal" href="#output-files-function-counts">4.3. Output files - function counts</a></li>
<li><a class="reference internal" href="#output-files-who-does-what-matrix">4.4. Output files - who does what matrix</a></li>
<li><a class="reference internal" href="#analysis-of-multiple-files-in-parallel-with-the-gui">4.5. Analysis of multiple files in parallel with the GUI</a></li>
<li><a class="reference internal" href="#analysis-of-e-coli-proteome">4.6. Analysis of <em>E. coli</em> proteome</a></li>
<li><a class="reference internal" href="#running-sequedex-from-the-command-line">4.7. Running Sequedex from the command line</a></li>
<li><a class="reference internal" href="#sequedex-configuration-file">4.8. Sequedex configuration file</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="installation.html"
                        title="previous chapter">3. Installation instructions</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="profiles.html"
                        title="next chapter">5. Phylogenetic and functional profiles</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/starting.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="profiles.html" title="5. Phylogenetic and functional profiles"
             >next</a> |</li>
        <li class="right" >
          <a href="installation.html" title="3. Installation instructions"
             >previous</a> |</li>
        <li><a href="index.html">Sequedex 0.9.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; <a href="copyright.html">Copyright</a> 2012, Los Alamos National Security, LLC.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>